// @spartacus/organization/esm2015/administration/components/user/user-groups/user-user-group-list.service.js
import { Injectable } from '@angular/core';
import { filter } from 'rxjs/operators';
import { SubListService } from '../../shared/sub-list/sub-list.service';
import { OrganizationTableType } from '../../shared/organization.model';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/storefront";
import * as i2 from "@spartacus/organization/administration/core";
export class UserUserGroupListService extends SubListService {
    constructor(tableService, userService, userGroupService) {
        super(tableService);
        this.tableService = tableService;
        this.userService = userService;
        this.userGroupService = userGroupService;
        this.tableType = OrganizationTableType.USER_USER_GROUPS;
        this._domainType = OrganizationTableType.USER_GROUP;
    }
    load(pagination, code) {
        return this.userService
            .getUserGroups(code, pagination)
            .pipe(filter((list) => Boolean(list)));
    }
    /**
     * @override
     * Assign user group to the user.
     */
    assign(userCode, userGroupCode) {
        this.userService.assignUserGroup(userCode, userGroupCode);
        return this.userGroupService.getLoadingStatus(userGroupCode);
    }
    /**
     * @override
     * Unassign the user group from the user.
     */
    unassign(userCode, userGroupCode) {
        this.userService.unassignUserGroup(userCode, userGroupCode);
        return this.userGroupService.getLoadingStatus(userGroupCode);
    }
}
UserUserGroupListService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserUserGroupListService, deps: [{ token: i1.TableService }, { token: i2.B2BUserService }, { token: i2.UserGroupService }], target: i0.ɵɵFactoryTarget.Injectable });
UserUserGroupListService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserUserGroupListService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserUserGroupListService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.TableService }, { type: i2.B2BUserService }, { type: i2.UserGroupService }]; } });
