// @spartacus/organization/esm2015/administration/components/user-group/permissions/user-group-permission-list.service.js
import { Injectable } from '@angular/core';
import { SubListService } from '../../shared/sub-list/sub-list.service';
import { OrganizationTableType } from '../../shared/organization.model';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/storefront";
import * as i2 from "@spartacus/organization/administration/core";
export class UserGroupPermissionListService extends SubListService {
    constructor(tableService, userGroupService, permissionService) {
        super(tableService);
        this.tableService = tableService;
        this.userGroupService = userGroupService;
        this.permissionService = permissionService;
        this.tableType = OrganizationTableType.USER_GROUP_PERMISSIONS;
        this._domainType = OrganizationTableType.PERMISSION;
    }
    /**
     *
     * @override
     * Loads all b2b users.
     *
     * @param code The user group code.
     */
    load(pagination, code) {
        return this.userGroupService.getAvailableOrderApprovalPermissions(code, pagination);
    }
    /**
     * @override
     * Assign user to the user group.
     */
    assign(userGroupCode, permissionCode) {
        this.userGroupService.assignPermission(userGroupCode, permissionCode);
        return this.permissionService.getLoadingStatus(permissionCode);
    }
    /**
     * @override
     * Unassigns the user from the user group.
     */
    unassign(userGroupCode, permissionCode) {
        this.userGroupService.unassignPermission(userGroupCode, permissionCode);
        return this.permissionService.getLoadingStatus(permissionCode);
    }
}
UserGroupPermissionListService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserGroupPermissionListService, deps: [{ token: i1.TableService }, { token: i2.UserGroupService }, { token: i2.PermissionService }], target: i0.ɵɵFactoryTarget.Injectable });
UserGroupPermissionListService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserGroupPermissionListService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserGroupPermissionListService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.TableService }, { type: i2.UserGroupService }, { type: i2.PermissionService }]; } });
