// @spartacus/organization/esm2015/administration/components/shared/item-active.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ItemActiveDirective } from './item-active.directive';
import * as i0 from "@angular/core";
export class ItemActiveModule {
}
ItemActiveModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ItemActiveModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ItemActiveModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ItemActiveModule, declarations: [ItemActiveDirective], imports: [CommonModule], exports: [ItemActiveDirective] });
ItemActiveModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ItemActiveModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ItemActiveModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [ItemActiveDirective],
                    exports: [ItemActiveDirective],
                }]
        }] });
