// @spartacus/organization/esm2015/administration/components/shared/item-active.directive.js
import { Directive } from '@angular/core';
import { GlobalMessageType } from '@spartacus/core';
import { distinctUntilChanged, filter } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "./item.service";
import * as i2 from "./message/services/message.service";
export class ItemActiveDirective {
    constructor(itemService, messageService) {
        this.itemService = itemService;
        this.messageService = messageService;
    }
    ngOnInit() {
        this.subscription = this.itemService.current$
            .pipe(distinctUntilChanged((previous, current) => (previous === null || previous === void 0 ? void 0 : previous.active) === (current === null || current === void 0 ? void 0 : current.active)), filter((item) => item && (item === null || item === void 0 ? void 0 : item.active) === false))
            .subscribe((item) => this.handleDisabledItems(item));
    }
    handleDisabledItems(item) {
        this.messageService.add({
            message: {
                key: 'organization.notification.disabled',
                params: { item },
            },
            type: GlobalMessageType.MSG_TYPE_ERROR,
        });
    }
    ngOnDestroy() {
        var _a;
        (_a = this.subscription) === null || _a === void 0 ? void 0 : _a.unsubscribe();
    }
}
ItemActiveDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ItemActiveDirective, deps: [{ token: i1.ItemService }, { token: i2.MessageService }], target: i0.ɵɵFactoryTarget.Directive });
ItemActiveDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.0.5", type: ItemActiveDirective, selector: "[cxOrgItemActive]", ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ItemActiveDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[cxOrgItemActive]',
                }]
        }], ctorParameters: function () { return [{ type: i1.ItemService }, { type: i2.MessageService }]; } });
