// @spartacus/organization/esm2015/administration/components/shared/item-exists.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ItemExistsDirective } from './item-exists.directive';
import * as i0 from "@angular/core";
export class ItemExistsModule {
}
ItemExistsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ItemExistsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ItemExistsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ItemExistsModule, declarations: [ItemExistsDirective], imports: [CommonModule], exports: [ItemExistsDirective] });
ItemExistsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ItemExistsModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ItemExistsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [ItemExistsDirective],
                    exports: [ItemExistsDirective],
                }]
        }] });
