// @spartacus/organization/esm2015/administration/components/shared/organization.model.js
export var OrganizationTableType;
(function (OrganizationTableType) {
    OrganizationTableType["BUDGET"] = "orgBudget";
    OrganizationTableType["BUDGET_ASSIGNED_COST_CENTERS"] = "orgBudgetAssignedCostCenters";
    OrganizationTableType["COST_CENTER"] = "orgCostCenter";
    OrganizationTableType["COST_CENTER_BUDGETS"] = "orgCostCenterBudgets";
    OrganizationTableType["COST_CENTER_ASSIGNED_BUDGETS"] = "orgCostCenterAssignedBudgets";
    OrganizationTableType["UNIT"] = "orgUnit";
    OrganizationTableType["UNIT_USERS"] = "orgUnitUsers";
    OrganizationTableType["UNIT_CHILDREN"] = "orgUnitChildren";
    OrganizationTableType["UNIT_APPROVERS"] = "orgUnitApprovers";
    OrganizationTableType["UNIT_ASSIGNED_APPROVERS"] = "orgUnitAssignedApprovers";
    OrganizationTableType["UNIT_ADDRESS"] = "orgUnitAddress";
    OrganizationTableType["UNIT_COST_CENTERS"] = "orgUnitCostCenters";
    OrganizationTableType["USER_GROUP"] = "orgUserGroup";
    OrganizationTableType["USER_GROUP_USERS"] = "orgUserGroupUsers";
    OrganizationTableType["USER_GROUP_ASSIGNED_USERS"] = "orgUserGroupAssignedUsers";
    OrganizationTableType["USER_GROUP_PERMISSIONS"] = "orgUserGroupPermissions";
    OrganizationTableType["USER_GROUP_ASSIGNED_PERMISSIONS"] = "orgUserGroupAssignedPermissions";
    OrganizationTableType["USER"] = "orgUser";
    OrganizationTableType["USER_APPROVERS"] = "orgUserApprovers";
    OrganizationTableType["USER_ASSIGNED_APPROVERS"] = "orgUserAssignedApprovers";
    OrganizationTableType["USER_PERMISSIONS"] = "orgUserPermissions";
    OrganizationTableType["USER_ASSIGNED_PERMISSIONS"] = "orgUserAssignedPermissions";
    OrganizationTableType["USER_USER_GROUPS"] = "orgUserUserGroups";
    OrganizationTableType["USER_ASSIGNED_USER_GROUPS"] = "orgUserAssignedUserGroups";
    OrganizationTableType["PERMISSION"] = "orgPurchaseLimit";
})(OrganizationTableType || (OrganizationTableType = {}));
