// @spartacus/organization/esm2015/administration/root/config/default-units-routing.config.js
import { ROUTE_PARAMS } from '../route-params';
const listPath = `organization/units/:${ROUTE_PARAMS.unitCode}`;
const paramsMapping = {
    unitCode: 'uid',
    addressId: 'id',
    userCode: 'customerId',
};
export const defaultUnitsRoutingConfig = {
    routing: {
        routes: {
            orgUnits: {
                paths: ['organization/units'],
            },
            orgUnitCreate: {
                paths: ['organization/units/create'],
            },
            orgUnitDetails: {
                paths: [listPath],
                paramsMapping,
            },
            orgUnitEdit: {
                paths: [`${listPath}/edit`],
                paramsMapping,
            },
            orgUnitChildren: {
                paths: [`${listPath}/children`],
                paramsMapping,
            },
            orgUnitCreateChild: {
                paths: [`${listPath}/children/create`],
                paramsMapping,
            },
            orgUnitUserList: {
                paths: [`${listPath}/users`],
                paramsMapping,
            },
            orgUnitCreateUser: {
                paths: [`${listPath}/users/create`],
                paramsMapping,
            },
            orgUnitUserRoles: {
                paths: [`${listPath}/users/:userCode/roles`],
                paramsMapping,
            },
            orgUnitApprovers: {
                paths: [`${listPath}/approvers`],
                paramsMapping,
            },
            orgUnitAssignApprovers: {
                paths: [`${listPath}/approvers/assign`],
                paramsMapping,
            },
            orgUnitAddressList: {
                paths: [`${listPath}/addresses`],
                paramsMapping,
            },
            orgUnitAddressCreate: {
                paths: [`${listPath}/addresses/create`],
                paramsMapping,
            },
            orgUnitAddressDetails: {
                paths: [`${listPath}/addresses/:addressId`],
                paramsMapping,
            },
            orgUnitAddressEdit: {
                paths: [`${listPath}/addresses/:addressId/edit`],
                paramsMapping,
            },
            orgUnitCostCenters: {
                paths: [`${listPath}/cost-centers`],
                paramsMapping,
            },
            orgUnitCreateCostCenter: {
                paths: [`${listPath}/cost-centers/create`],
                paramsMapping,
            },
        },
    },
};
