// @spartacus/organization/esm2015/administration/root/config/default-cost-center-routing.config.js
import { ROUTE_PARAMS } from '../route-params';
const listPath = `organization/cost-centers/:${ROUTE_PARAMS.costCenterCode}`;
const paramsMapping = {
    costCenterCode: 'code',
};
export const defaultCostCenterRoutingConfig = {
    routing: {
        routes: {
            orgCostCenter: {
                paths: ['organization/cost-centers'],
            },
            orgCostCenterCreate: {
                paths: ['organization/cost-centers/create'],
            },
            orgCostCenterDetails: {
                paths: [`${listPath}`],
                paramsMapping,
            },
            orgCostCenterBudgets: {
                paths: [`${listPath}/budgets`],
                paramsMapping,
            },
            orgCostCenterAssignBudgets: {
                paths: [`${listPath}/budgets/assign`],
                paramsMapping,
            },
            orgCostCenterEdit: {
                paths: [`${listPath}/edit`],
                paramsMapping,
            },
        },
    },
};
