// @spartacus/organization/esm2015/administration/root/administration-root.module.js
import { NgModule } from '@angular/core';
import { HttpErrorHandler, provideDefaultConfig, provideDefaultConfigFactory, } from '@spartacus/core';
import { OrganizationBadRequestHandler, OrganizationConflictHandler, } from '@spartacus/organization/administration/core';
import { defaultBudgetRoutingConfig } from './config/default-budget-routing.config';
import { defaultCostCenterRoutingConfig } from './config/default-cost-center-routing.config';
import { defaultOrganizationLayoutConfig } from './config/default-organization-layout.config';
import { defaultPermissionRoutingConfig } from './config/default-permission-routing.config';
import { defaultUnitsRoutingConfig } from './config/default-units-routing.config';
import { defaultUserGroupRoutingConfig } from './config/default-user-group-routing.config';
import { defaultUserRoutingConfig } from './config/default-user-routing.config';
import { ORGANIZATION_ADMINISTRATION_FEATURE } from './feature-name';
import * as i0 from "@angular/core";
// TODO: Inline this factory when we start releasing Ivy compiled libraries
export function defaultOrganizationAdministrationComponentsConfig() {
    const config = {
        featureModules: {
            [ORGANIZATION_ADMINISTRATION_FEATURE]: {
                cmsComponents: [
                    'ManageBudgetsListComponent',
                    'ManageCostCentersListComponent',
                    'ManagePermissionsListComponent',
                    'ManageUnitsListComponent',
                    'ManageUsersListComponent',
                    'ManageUserGroupsListComponent',
                ],
            },
        },
    };
    return config;
}
export class AdministrationRootModule {
}
AdministrationRootModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AdministrationRootModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AdministrationRootModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AdministrationRootModule });
AdministrationRootModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AdministrationRootModule, providers: [
        provideDefaultConfig(defaultOrganizationLayoutConfig),
        provideDefaultConfig({
            routing: { routes: { organization: { paths: ['organization'] } } },
        }),
        provideDefaultConfig(defaultBudgetRoutingConfig),
        provideDefaultConfig(defaultCostCenterRoutingConfig),
        provideDefaultConfig(defaultPermissionRoutingConfig),
        provideDefaultConfig(defaultUnitsRoutingConfig),
        provideDefaultConfig(defaultUserRoutingConfig),
        provideDefaultConfig(defaultUserGroupRoutingConfig),
        provideDefaultConfigFactory(defaultOrganizationAdministrationComponentsConfig),
        {
            provide: HttpErrorHandler,
            useExisting: OrganizationConflictHandler,
            multi: true,
        },
        {
            provide: HttpErrorHandler,
            useExisting: OrganizationBadRequestHandler,
            multi: true,
        },
    ] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: AdministrationRootModule, decorators: [{
            type: NgModule,
            args: [{
                    providers: [
                        provideDefaultConfig(defaultOrganizationLayoutConfig),
                        provideDefaultConfig({
                            routing: { routes: { organization: { paths: ['organization'] } } },
                        }),
                        provideDefaultConfig(defaultBudgetRoutingConfig),
                        provideDefaultConfig(defaultCostCenterRoutingConfig),
                        provideDefaultConfig(defaultPermissionRoutingConfig),
                        provideDefaultConfig(defaultUnitsRoutingConfig),
                        provideDefaultConfig(defaultUserRoutingConfig),
                        provideDefaultConfig(defaultUserGroupRoutingConfig),
                        provideDefaultConfigFactory(defaultOrganizationAdministrationComponentsConfig),
                        {
                            provide: HttpErrorHandler,
                            useExisting: OrganizationConflictHandler,
                            multi: true,
                        },
                        {
                            provide: HttpErrorHandler,
                            useExisting: OrganizationBadRequestHandler,
                            multi: true,
                        },
                    ],
                }]
        }] });
