// @spartacus/organization/esm2015/administration/components/unit/services/unit-list.service.js
import { Injectable } from '@angular/core';
import { map, switchMap } from 'rxjs/operators';
import { ListService } from '../../shared/list/list.service';
import { OrganizationTableType } from '../../shared/organization.model';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/storefront";
import * as i2 from "@spartacus/organization/administration/core";
import * as i3 from "./unit-item.service";
import * as i4 from "./unit-tree.service";
/**
 * Service to populate Unit data to `Table` data. Unit
 * data is driven by the table configuration, using the `OrganizationTables.UNIT`.
 */
export class UnitListService extends ListService {
    constructor(tableService, unitService, unitItemService, unitTreeService) {
        super(tableService);
        this.tableService = tableService;
        this.unitService = unitService;
        this.unitItemService = unitItemService;
        this.unitTreeService = unitTreeService;
        this.tableType = OrganizationTableType.UNIT;
    }
    load() {
        return this.unitService.getTree().pipe(switchMap((node) => this.unitItemService.key$.pipe(map((key) => {
            this.unitTreeService.initialize(node, key);
            return node;
        }))), switchMap((tree) => this.unitTreeService.treeToggle$.pipe(map(() => tree))), map((tree) => this.convertListItem(tree)));
    }
    convertListItem(unit, depthLevel = 0, pagination = { totalResults: 0 }) {
        var _a, _b;
        let values = [];
        if (!unit) {
            return;
        }
        const node = Object.assign(Object.assign({}, unit), { count: (_b = (_a = unit.children) === null || _a === void 0 ? void 0 : _a.length) !== null && _b !== void 0 ? _b : 0, expanded: this.unitTreeService.isExpanded(unit.id, depthLevel), depthLevel, 
            // tmp, should be normalized
            uid: unit.id, children: [...unit.children].sort((unitA, unitB) => unitA.name.localeCompare(unitB.name)) });
        values.push(node);
        pagination.totalResults++;
        node.children.forEach((childUnit) => {
            var _a;
            const childList = (_a = this.convertListItem(childUnit, depthLevel + 1, pagination)) === null || _a === void 0 ? void 0 : _a.values;
            if (node.expanded && childList.length > 0) {
                values = values.concat(childList);
            }
        });
        return { values, pagination };
    }
    key() {
        return 'uid';
    }
}
UnitListService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitListService, deps: [{ token: i1.TableService }, { token: i2.OrgUnitService }, { token: i3.UnitItemService }, { token: i4.UnitTreeService }], target: i0.ɵɵFactoryTarget.Injectable });
UnitListService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitListService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitListService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.TableService }, { type: i2.OrgUnitService }, { type: i3.UnitItemService }, { type: i4.UnitTreeService }]; } });
