// @spartacus/organization/esm2015/administration/components/shared/list/list.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { NgSelectModule } from '@ng-select/ng-select';
import { I18nModule, UrlModule } from '@spartacus/core';
import { IconModule, KeyboardFocusModule, PaginationModule, SplitViewModule, TableModule, } from '@spartacus/storefront';
import { MessageModule } from '../message/message.module';
import { ListComponent } from './list.component';
import { PopoverModule } from '@spartacus/storefront';
import * as i0 from "@angular/core";
export class ListModule {
}
ListModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ListModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ListModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ListModule, declarations: [ListComponent], imports: [CommonModule,
        RouterModule,
        SplitViewModule,
        TableModule,
        IconModule,
        UrlModule,
        I18nModule,
        PaginationModule,
        NgSelectModule,
        FormsModule,
        MessageModule,
        KeyboardFocusModule,
        PopoverModule], exports: [ListComponent] });
ListModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ListModule, imports: [[
            CommonModule,
            RouterModule,
            SplitViewModule,
            TableModule,
            IconModule,
            UrlModule,
            I18nModule,
            PaginationModule,
            NgSelectModule,
            FormsModule,
            MessageModule,
            KeyboardFocusModule,
            PopoverModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ListModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        RouterModule,
                        SplitViewModule,
                        TableModule,
                        IconModule,
                        UrlModule,
                        I18nModule,
                        PaginationModule,
                        NgSelectModule,
                        FormsModule,
                        MessageModule,
                        KeyboardFocusModule,
                        PopoverModule,
                    ],
                    declarations: [ListComponent],
                    exports: [ListComponent],
                }]
        }] });
