// @spartacus/organization/esm2015/administration/components/shared/list/list.service.js
import { Injectable } from '@angular/core';
import { TableLayout, } from '@spartacus/storefront';
import { BehaviorSubject, Subject } from 'rxjs';
import { map, startWith, switchMap } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/storefront";
/**
 * The `ListService` deals with the table structure, list data and
 * pagination of tables inside the b2b organization.
 *
 * @property {OrganizationTableType} tableType
 *   Used to load the table structure configuration and generate table outlets.
 * @property {PaginationModel} pagination$
 *   The pagination state of the listing.
 */
export class ListService {
    constructor(tableService) {
        this.tableService = tableService;
        /**
         * The default table structure is used to add the default configuration for all
         * organization list related tables. This avoids a lot of boilerplate configuration.
         */
        this.defaultTableStructure = {
            options: { layout: TableLayout.VERTICAL_STACKED },
            lg: { options: { layout: TableLayout.VERTICAL } },
        };
        /**
         * The ghost data contains an empty list of objects that is used in the UI
         * to render the HTML elements.
         *
         * This list contains 10 items, so that the ghost will show 10 rows by default.
         */
        this.ghostData = { values: new Array(10) };
        this.notification$ = new Subject();
        /**
         * The pagination state of the listing.
         *
         * The pagination size defaults to 10, but can be overridden by the
         * table configuration for each entity type.
         */
        this.pagination$ = new BehaviorSubject({
            pageSize: 10,
        });
    }
    get viewType() {
        return this.tableType;
    }
    get domainType() {
        var _a;
        return (_a = this._domainType) !== null && _a !== void 0 ? _a : this.viewType;
    }
    /**
     * Indicates the unique key for the item model. The key is different for various
     * organizations, i.e. `budget.code`, `user.uid`.
     */
    key() {
        return 'code';
    }
    /**
     * Loads the data by delegating to the `load` method, which must be implemented
     * in specific implementations of this abstract class.
     *
     * The load method is streamed from the `pagination$` stream, which is initialized
     * with default pagination and structure drive properties.
     */
    getData(...args) {
        return this.pagination$.pipe(
        // we merge any configured pagination from the table structure
        switchMap((pagination) => this.getStructure().pipe(map((config) => { var _a; return (Object.assign(Object.assign({}, pagination), (_a = config.options) === null || _a === void 0 ? void 0 : _a.pagination)); }))), switchMap((pagination) => this.load(pagination, ...args)), startWith(this.ghostData));
    }
    /**
     * Returns the `TableStructure` for the `OrganizationTableType`.
     *
     * The table structure is build by the `TableService` based on configuration.
     * The `defaultTableStructure` is deep merged as a fallback configuration.
     */
    getStructure() {
        return this.tableService.buildStructure(this.viewType, this.defaultTableStructure);
    }
    /**
     * Views the page.
     */
    view(pagination, nextPage) {
        this.pagination$.next(Object.assign(Object.assign({}, pagination), { currentPage: nextPage }));
    }
    /**
     * Updates the sort code for the PaginationModel.
     *
     * The `currentPage` is reset to 0.
     */
    sort(pagination, _obsoleteSort) {
        this.view(pagination, 0);
    }
    /**
     * Indicates whether the given data equals to the ghost data.
     *
     * This is used to validate the initial loading state, which is
     * different from the loading state; the loading state occurs
     * while sorting and paginating, where as the initial loading state
     * only happens at the very first load.
     */
    hasGhostData(data) {
        return data === this.ghostData;
    }
}
ListService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ListService, deps: [{ token: i1.TableService }], target: i0.ɵɵFactoryTarget.Injectable });
ListService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ListService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ListService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.TableService }]; } });
