// @spartacus/organization/esm2015/administration/components/permission/details-cell/permission-details-cell.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { I18nModule, UrlModule } from '@spartacus/core';
import { PopoverModule } from '@spartacus/storefront';
import { PermissionDetailsCellComponent } from './permission-details-cell.component';
import * as i0 from "@angular/core";
export class PermissionDetailsCellModule {
}
PermissionDetailsCellModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PermissionDetailsCellModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
PermissionDetailsCellModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PermissionDetailsCellModule, declarations: [PermissionDetailsCellComponent], imports: [CommonModule, PopoverModule, RouterModule, I18nModule, UrlModule], exports: [PermissionDetailsCellComponent] });
PermissionDetailsCellModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PermissionDetailsCellModule, imports: [[CommonModule, PopoverModule, RouterModule, I18nModule, UrlModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PermissionDetailsCellModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, PopoverModule, RouterModule, I18nModule, UrlModule],
                    declarations: [PermissionDetailsCellComponent],
                    exports: [PermissionDetailsCellComponent],
                }]
        }] });
