// @spartacus/organization/esm2015/order-approval/components/details/order-approval-details.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { AuthGuard, ConfigModule, I18nModule, UrlModule, } from '@spartacus/core';
import { CartSharedModule, FormErrorsModule, OrderDetailItemsComponent, OrderDetailShippingComponent, OrderDetailsService, OrderDetailTotalsComponent, SpinnerModule, } from '@spartacus/storefront';
import { ApproverGuard } from '../../core/guards/approver.guard';
import { OrderApprovalDetailFormComponent } from './order-approval-detail-form/order-approval-detail-form.component';
import { OrderApprovalDetailService } from './order-approval-detail.service';
import { OrderDetailPermissionResultsComponent } from './order-detail-permission-results/order-detail-permission-results.component';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
export class OrderApprovalDetailsModule {
}
OrderApprovalDetailsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderApprovalDetailsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
OrderApprovalDetailsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderApprovalDetailsModule, declarations: [OrderApprovalDetailFormComponent,
        OrderDetailPermissionResultsComponent], imports: [ReactiveFormsModule,
        CartSharedModule,
        CommonModule,
        I18nModule,
        UrlModule,
        FormErrorsModule,
        SpinnerModule,
        RouterModule, i1.ConfigModule], exports: [OrderApprovalDetailFormComponent,
        OrderDetailPermissionResultsComponent] });
OrderApprovalDetailsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderApprovalDetailsModule, imports: [[
            ReactiveFormsModule,
            CartSharedModule,
            CommonModule,
            I18nModule,
            UrlModule,
            FormErrorsModule,
            SpinnerModule,
            RouterModule,
            ConfigModule.withConfig({
                cmsComponents: {
                    OrderApprovalDetailTotalsComponent: {
                        component: OrderDetailTotalsComponent,
                        providers: [
                            {
                                provide: OrderDetailsService,
                                useExisting: OrderApprovalDetailService,
                            },
                        ],
                        guards: [AuthGuard, ApproverGuard],
                    },
                    OrderApprovalDetailApprovalDetailsComponent: {
                        component: OrderDetailPermissionResultsComponent,
                        providers: [
                            {
                                provide: OrderDetailsService,
                                useExisting: OrderApprovalDetailService,
                            },
                        ],
                        guards: [AuthGuard, ApproverGuard],
                    },
                    AccountOrderDetailsApprovalDetailsComponent: {
                        component: OrderDetailPermissionResultsComponent,
                    },
                    OrderApprovalDetailShippingComponent: {
                        component: OrderDetailShippingComponent,
                        providers: [
                            {
                                provide: OrderDetailsService,
                                useExisting: OrderApprovalDetailService,
                            },
                        ],
                        guards: [AuthGuard, ApproverGuard],
                    },
                    OrderApprovalDetailItemsComponent: {
                        component: OrderDetailItemsComponent,
                        providers: [
                            {
                                provide: OrderDetailsService,
                                useExisting: OrderApprovalDetailService,
                            },
                        ],
                        guards: [AuthGuard, ApproverGuard],
                    },
                    OrderApprovalDetailFormComponent: {
                        component: OrderApprovalDetailFormComponent,
                        guards: [AuthGuard, ApproverGuard],
                    },
                },
            }),
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderApprovalDetailsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        ReactiveFormsModule,
                        CartSharedModule,
                        CommonModule,
                        I18nModule,
                        UrlModule,
                        FormErrorsModule,
                        SpinnerModule,
                        RouterModule,
                        ConfigModule.withConfig({
                            cmsComponents: {
                                OrderApprovalDetailTotalsComponent: {
                                    component: OrderDetailTotalsComponent,
                                    providers: [
                                        {
                                            provide: OrderDetailsService,
                                            useExisting: OrderApprovalDetailService,
                                        },
                                    ],
                                    guards: [AuthGuard, ApproverGuard],
                                },
                                OrderApprovalDetailApprovalDetailsComponent: {
                                    component: OrderDetailPermissionResultsComponent,
                                    providers: [
                                        {
                                            provide: OrderDetailsService,
                                            useExisting: OrderApprovalDetailService,
                                        },
                                    ],
                                    guards: [AuthGuard, ApproverGuard],
                                },
                                AccountOrderDetailsApprovalDetailsComponent: {
                                    component: OrderDetailPermissionResultsComponent,
                                },
                                OrderApprovalDetailShippingComponent: {
                                    component: OrderDetailShippingComponent,
                                    providers: [
                                        {
                                            provide: OrderDetailsService,
                                            useExisting: OrderApprovalDetailService,
                                        },
                                    ],
                                    guards: [AuthGuard, ApproverGuard],
                                },
                                OrderApprovalDetailItemsComponent: {
                                    component: OrderDetailItemsComponent,
                                    providers: [
                                        {
                                            provide: OrderDetailsService,
                                            useExisting: OrderApprovalDetailService,
                                        },
                                    ],
                                    guards: [AuthGuard, ApproverGuard],
                                },
                                OrderApprovalDetailFormComponent: {
                                    component: OrderApprovalDetailFormComponent,
                                    guards: [AuthGuard, ApproverGuard],
                                },
                            },
                        }),
                    ],
                    declarations: [
                        OrderApprovalDetailFormComponent,
                        OrderDetailPermissionResultsComponent,
                    ],
                    exports: [
                        OrderApprovalDetailFormComponent,
                        OrderDetailPermissionResultsComponent,
                    ],
                }]
        }] });
