// @spartacus/organization/esm2015/administration/core/connectors/permission/permission.connector.js
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./permission.adapter";
export class PermissionConnector {
    constructor(adapter) {
        this.adapter = adapter;
    }
    get(userId, permissionCode) {
        return this.adapter.load(userId, permissionCode);
    }
    getList(userId, params) {
        return this.adapter.loadList(userId, params);
    }
    create(userId, permission) {
        return this.adapter.create(userId, permission);
    }
    update(userId, permissionCode, permission) {
        return this.adapter.update(userId, permissionCode, permission);
    }
    getTypes() {
        return this.adapter.loadTypes();
    }
}
PermissionConnector.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PermissionConnector, deps: [{ token: i1.PermissionAdapter }], target: i0.ɵɵFactoryTarget.Injectable });
PermissionConnector.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PermissionConnector, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PermissionConnector, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.PermissionAdapter }]; } });
