// @spartacus/organization/esm2015/administration/components/unit/links/approvers/assigned/unit-assigned-approver-list.service.js
import { Injectable } from '@angular/core';
import { B2BUserRole, } from '@spartacus/core';
import { map } from 'rxjs/operators';
import { OrganizationTableType } from '../../../../shared/index';
import { UnitApproverListService } from '../unit-approver-list.service';
import * as i0 from "@angular/core";
export class UnitAssignedApproverListService extends UnitApproverListService {
    constructor() {
        super(...arguments);
        this.tableType = OrganizationTableType.UNIT_ASSIGNED_APPROVERS;
    }
    load(pagination, code) {
        this.unitService.clearAssignedUsersList(code, B2BUserRole.APPROVER, pagination);
        return super
            .load(pagination, code)
            .pipe(map((users) => this.filterSelected(users)));
    }
}
UnitAssignedApproverListService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitAssignedApproverListService, deps: null, target: i0.ɵɵFactoryTarget.Injectable });
UnitAssignedApproverListService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitAssignedApproverListService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitAssignedApproverListService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }] });
