// @spartacus/organization/esm2015/administration/components/unit/links/users/roles/unit-user-roles-item.service.js
import { Injectable } from '@angular/core';
import { ItemService } from '../../../../shared/item.service';
import * as i0 from "@angular/core";
import * as i1 from "../services/current-unit-user.service";
import * as i2 from "@spartacus/core";
import * as i3 from "./unit-user-roles-form.service";
import * as i4 from "@spartacus/organization/administration/core";
export class UnitUserRolesItemService extends ItemService {
    constructor(currentItemService, routingService, formService, unitService, b2bUserService) {
        super(currentItemService, routingService, formService);
        this.currentItemService = currentItemService;
        this.routingService = routingService;
        this.formService = formService;
        this.unitService = unitService;
        this.b2bUserService = b2bUserService;
    }
    save(form, key) {
        // we enable the unit so that the underlying
        // save method can read the complete form.value.
        form.get('orgUnit').enable();
        return super.save(form, key);
    }
    load(unitUid) {
        return this.b2bUserService.get(unitUid);
    }
    update(customerId, _user) {
        return this.b2bUserService.getLoadingStatus(customerId);
    }
    create(_customer) {
        return this.b2bUserService.getLoadingStatus(null);
    }
    getDetailsRoute() {
        return this.currentItemService.getDetailsRoute();
    }
}
UnitUserRolesItemService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitUserRolesItemService, deps: [{ token: i1.CurrentUnitUserService }, { token: i2.RoutingService }, { token: i3.UnitUserRolesFormService }, { token: i4.OrgUnitService }, { token: i4.B2BUserService }], target: i0.ɵɵFactoryTarget.Injectable });
UnitUserRolesItemService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitUserRolesItemService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitUserRolesItemService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.CurrentUnitUserService }, { type: i2.RoutingService }, { type: i3.UnitUserRolesFormService }, { type: i4.OrgUnitService }, { type: i4.B2BUserService }]; } });
