// @spartacus/organization/esm2015/administration/components/unit/links/users/roles/unit-user-roles-form.service.js
import { Injectable } from '@angular/core';
import { FormControl, FormGroup } from '@angular/forms';
import { FormService } from '../../../../shared/form/form.service';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/organization/administration/core";
export class UnitUserRolesFormService extends FormService {
    constructor(userService) {
        super();
        this.userService = userService;
        this.availableRoles = this.userService.getAllRoles();
    }
    getForm(item) {
        // if form already exist, while switching between users
        // it didn't patchData again, so used force rebuild
        this.form = null;
        return super.getForm(item);
    }
    build() {
        const form = new FormGroup({});
        this.availableRoles.forEach((role) => form.addControl(role, new FormControl()));
        this.form = form;
    }
    patchData(item) {
        var _a;
        super.patchData(item);
        if (item) {
            (_a = item.roles) === null || _a === void 0 ? void 0 : _a.forEach((role) => {
                this.form.get(role).setValue(true);
            });
        }
    }
}
UnitUserRolesFormService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitUserRolesFormService, deps: [{ token: i1.B2BUserService }], target: i0.ɵɵFactoryTarget.Injectable });
UnitUserRolesFormService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitUserRolesFormService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitUserRolesFormService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.B2BUserService }]; } });
