// @spartacus/organization/esm2015/administration/components/unit/links/cost-centers/create/unit-cost-center-create.component.js
import { ChangeDetectionStrategy, Component } from '@angular/core';
import { CostCenterItemService } from '../../../../cost-center/services/cost-center-item.service';
import { UnitCostCenterItemService } from './unit-cost-center-item.service';
import * as i0 from "@angular/core";
import * as i1 from "../../../services/current-unit.service";
import * as i2 from "../../../../cost-center/form/cost-center-form.component";
import * as i3 from "@angular/common";
export class UnitCostCenterCreateComponent {
    constructor(unitService) {
        this.unitService = unitService;
        this.unitKey$ = this.unitService.key$;
    }
}
UnitCostCenterCreateComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitCostCenterCreateComponent, deps: [{ token: i1.CurrentUnitService }], target: i0.ɵɵFactoryTarget.Component });
UnitCostCenterCreateComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: UnitCostCenterCreateComponent, selector: "cx-org-unit-cost-center-create", host: { classAttribute: "content-wrapper" }, providers: [
        // we provide a specific version of the `CostCenterItemService` to
        // let the form component work with unit cost centers.
        {
            provide: CostCenterItemService,
            useExisting: UnitCostCenterItemService,
        },
    ], ngImport: i0, template: "<cx-org-cost-center-form [unitKey]=\"unitKey$ | async\"></cx-org-cost-center-form>\n", components: [{ type: i2.CostCenterFormComponent, selector: "cx-org-cost-center-form", inputs: ["unitKey"] }], pipes: { "async": i3.AsyncPipe }, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitCostCenterCreateComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'cx-org-unit-cost-center-create',
                    templateUrl: './unit-cost-center-create.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    host: { class: 'content-wrapper' },
                    providers: [
                        // we provide a specific version of the `CostCenterItemService` to
                        // let the form component work with unit cost centers.
                        {
                            provide: CostCenterItemService,
                            useExisting: UnitCostCenterItemService,
                        },
                    ],
                }]
        }], ctorParameters: function () { return [{ type: i1.CurrentUnitService }]; } });
