// @spartacus/organization/esm2015/administration/components/unit/links/cost-centers/create/unit-cost-center-item.service.js
import { Injectable } from '@angular/core';
import { CostCenterItemService } from '../../../../cost-center/services/cost-center-item.service';
import * as i0 from "@angular/core";
export class UnitCostCenterItemService extends CostCenterItemService {
    save(form, key) {
        // we enable the unit so that the underlying
        // save method can read the complete form.value.
        form.get('unit').enable();
        return super.save(form, key);
    }
    /**
     * @override
     * Returns 'orgUnitCostCenters'
     */
    getDetailsRoute() {
        return 'orgUnitCostCenters';
    }
    buildRouteParams(item) {
        return { uid: item.unit.uid };
    }
}
UnitCostCenterItemService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitCostCenterItemService, deps: null, target: i0.ɵɵFactoryTarget.Injectable });
UnitCostCenterItemService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitCostCenterItemService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitCostCenterItemService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }] });
