// @spartacus/organization/esm2015/administration/components/user/permissions/user-permission-list.service.js
import { Injectable } from '@angular/core';
import { SubListService } from '../../shared/sub-list/sub-list.service';
import { OrganizationTableType } from '../../shared/organization.model';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/storefront";
import * as i2 from "@spartacus/organization/administration/core";
export class UserPermissionListService extends SubListService {
    constructor(tableService, userService, permissionService) {
        super(tableService);
        this.tableService = tableService;
        this.userService = userService;
        this.permissionService = permissionService;
        this.tableType = OrganizationTableType.USER_PERMISSIONS;
        this._domainType = OrganizationTableType.PERMISSION;
    }
    load(pagination, code) {
        return this.userService.getPermissions(code, pagination);
    }
    /**
     * @override
     * Assign permission to the user.
     */
    assign(userCode, code) {
        this.userService.assignPermission(userCode, code);
        return this.permissionService.getLoadingStatus(code);
    }
    /**
     * @override
     * Unassign the permission from the user.
     */
    unassign(userCode, code) {
        this.userService.unassignPermission(userCode, code);
        return this.permissionService.getLoadingStatus(code);
    }
}
UserPermissionListService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserPermissionListService, deps: [{ token: i1.TableService }, { token: i2.B2BUserService }, { token: i2.PermissionService }], target: i0.ɵɵFactoryTarget.Injectable });
UserPermissionListService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserPermissionListService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserPermissionListService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.TableService }, { type: i2.B2BUserService }, { type: i2.PermissionService }]; } });
