// @spartacus/organization/esm2015/administration/components/shared/form/form.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { I18nModule } from '@spartacus/core';
import { ItemActiveModule } from '../item-active.module';
import { CardModule } from '../card/card.module';
import { MessageModule } from '../message/message.module';
import { MessageService } from '../message/services/message.service';
import { FormComponent } from './form.component';
import { KeyboardFocusModule } from '@spartacus/storefront';
import * as i0 from "@angular/core";
export class FormModule {
}
FormModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: FormModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
FormModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: FormModule, declarations: [FormComponent], imports: [CommonModule,
        FormsModule,
        I18nModule,
        RouterModule,
        CardModule,
        MessageModule,
        ItemActiveModule,
        KeyboardFocusModule], exports: [FormComponent] });
FormModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: FormModule, providers: [MessageService], imports: [[
            CommonModule,
            FormsModule,
            I18nModule,
            RouterModule,
            CardModule,
            MessageModule,
            ItemActiveModule,
            KeyboardFocusModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: FormModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        FormsModule,
                        I18nModule,
                        RouterModule,
                        CardModule,
                        MessageModule,
                        ItemActiveModule,
                        KeyboardFocusModule,
                    ],
                    declarations: [FormComponent],
                    providers: [MessageService],
                    exports: [FormComponent],
                }]
        }] });
