// @spartacus/organization/esm2015/administration/components/shared/form/form.service.js
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
export class FormService {
    getForm(item) {
        if (this.form && !!item) {
            this.patchData(item);
            return this.form;
        }
        if (!this.form) {
            this.build(item);
        }
        // while we should be able to reset with initial value, this doesn't always work
        // hence, we're patching afterwards.
        this.form.reset();
        this.form.enable();
        this.patchData(item);
        return this.form;
    }
    patchData(item) {
        this.toggleFreeze(item);
        this.form.patchValue(Object.assign(Object.assign({}, this.defaultValue), item));
    }
    toggleFreeze(item) {
        var _a, _b;
        if (this.form.enabled && ((_a = item) === null || _a === void 0 ? void 0 : _a.active) === false) {
            this.form.disable();
        }
        if (this.form.disabled && ((_b = item) === null || _b === void 0 ? void 0 : _b.active) === true) {
            this.form.enable();
        }
    }
    /**
     * returns the default form value.
     */
    get defaultValue() {
        return {};
    }
}
FormService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: FormService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
FormService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: FormService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: FormService, decorators: [{
            type: Injectable
        }] });
