// @spartacus/organization/schematics/add-organization/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.addSpartacusOrganization = void 0;
const schematics_1 = require("@angular-devkit/schematics");
const schematics_2 = require("@spartacus/schematics");
const package_json_1 = require("../../package.json");
const constants_1 = require("../constants");
function addSpartacusOrganization(options) {
    return (tree, _context) => {
        const packageJson = schematics_2.readPackageJson(tree);
        schematics_2.validateSpartacusInstallation(packageJson);
        return schematics_1.chain([
            schematics_2.addPackageJsonDependenciesForLibrary(package_json_1.peerDependencies, options),
            schematics_2.shouldAddFeature(schematics_2.CLI_ORGANIZATION_ADMINISTRATION_FEATURE, options.features)
                ? schematics_1.chain([
                    addAdministrationFeature(options),
                    schematics_2.configureB2bFeatures(options, packageJson),
                ])
                : schematics_1.noop(),
            schematics_2.shouldAddFeature(schematics_2.CLI_ORGANIZATION_ORDER_APPROVAL_FEATURE, options.features)
                ? schematics_1.chain([
                    addOrderApprovalsFeature(options),
                    schematics_2.configureB2bFeatures(options, packageJson),
                ])
                : schematics_1.noop(),
        ]);
    };
}
exports.addSpartacusOrganization = addSpartacusOrganization;
function addAdministrationFeature(options) {
    return schematics_2.addLibraryFeature(options, {
        folderName: constants_1.ORGANIZATION_FOLDER_NAME,
        moduleName: constants_1.ORGANIZATION_ADMINISTRATION_MODULE_NAME,
        featureModule: {
            name: constants_1.ADMINISTRATION_MODULE,
            importPath: constants_1.SPARTACUS_ADMINISTRATION,
        },
        rootModule: {
            name: constants_1.ADMINISTRATION_ROOT_MODULE,
            importPath: constants_1.SPARTACUS_ADMINISTRATION_ROOT,
        },
        lazyLoadingChunk: {
            moduleSpecifier: constants_1.SPARTACUS_ADMINISTRATION_ROOT,
            namedImports: [constants_1.ORGANIZATION_ADMINISTRATION_FEATURE_NAME_CONSTANT],
        },
        i18n: {
            resources: constants_1.ORGANIZATION_TRANSLATIONS,
            chunks: constants_1.ORGANIZATION_TRANSLATION_CHUNKS_CONFIG,
            importPath: constants_1.SPARTACUS_ADMINISTRATION_ASSETS,
        },
        styles: {
            scssFileName: constants_1.SCSS_FILE_NAME,
            importStyle: schematics_2.SPARTACUS_ORGANIZATION,
        },
    });
}
function addOrderApprovalsFeature(options) {
    return schematics_2.addLibraryFeature(options, {
        folderName: constants_1.ORGANIZATION_FOLDER_NAME,
        moduleName: constants_1.ORGANIZATION_ORDER_APPROVAL_MODULE_NAME,
        featureModule: {
            name: constants_1.ORDER_APPROVAL_MODULE,
            importPath: constants_1.SPARTACUS_ORDER_APPROVAL,
        },
        rootModule: {
            name: constants_1.ORDER_APPROVAL_ROOT_MODULE,
            importPath: constants_1.SPARTACUS_ORDER_APPROVAL_ROOT,
        },
        lazyLoadingChunk: {
            moduleSpecifier: constants_1.SPARTACUS_ORDER_APPROVAL_ROOT,
            namedImports: [constants_1.ORGANIZATION_ORDER_APPROVAL_FEATURE_NAME_CONSTANT],
        },
        i18n: {
            resources: constants_1.ORDER_APPROVAL_TRANSLATIONS,
            chunks: constants_1.ORDER_APPROVAL_TRANSLATION_CHUNKS_CONFIG,
            importPath: constants_1.SPARTACUS_ORDER_APPROVAL_ASSETS,
        },
        styles: {
            scssFileName: constants_1.SCSS_FILE_NAME,
            importStyle: schematics_2.SPARTACUS_ORGANIZATION,
        },
    });
}
