// @spartacus/organization/esm2015/administration/components/unit/links/approvers/unit-approver-list.service.js
import { Injectable } from '@angular/core';
import { B2BUserRole, } from '@spartacus/core';
import { SubListService } from '../../../shared/sub-list/sub-list.service';
import { OrganizationTableType } from '../../../shared/organization.model';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/storefront";
import * as i2 from "@spartacus/organization/administration/core";
export class UnitApproverListService extends SubListService {
    constructor(tableService, unitService, userService) {
        super(tableService);
        this.tableService = tableService;
        this.unitService = unitService;
        this.userService = userService;
        this.tableType = OrganizationTableType.UNIT_APPROVERS;
        this._domainType = OrganizationTableType.USER;
    }
    load(pagination, code) {
        return this.unitService.getUsers(code, B2BUserRole.APPROVER, pagination);
    }
    /**
     * @override
     * Assign budget to the cost center.
     */
    assign(unitId, customerId) {
        this.unitService.assignApprover(unitId, customerId, B2BUserRole.APPROVER);
        return this.userService.getLoadingStatus(customerId);
    }
    /**
     * @override
     * Unassign the budget from the cost center.
     */
    unassign(unitId, customerId) {
        this.unitService.unassignApprover(unitId, customerId, B2BUserRole.APPROVER);
        return this.userService.getLoadingStatus(customerId);
    }
}
UnitApproverListService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitApproverListService, deps: [{ token: i1.TableService }, { token: i2.OrgUnitService }, { token: i2.B2BUserService }], target: i0.ɵɵFactoryTarget.Injectable });
UnitApproverListService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitApproverListService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitApproverListService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.TableService }, { type: i2.OrgUnitService }, { type: i2.B2BUserService }]; } });
