// @spartacus/organization/esm2015/administration/core/services/organization-page-meta.module.js
import { NgModule } from '@angular/core';
import { PageMetaResolver } from '@spartacus/core';
import { OrganizationPageMetaResolver } from './organization-page-meta.resolver';
import * as i0 from "@angular/core";
export class OrganizationPageMetaModule {
}
OrganizationPageMetaModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrganizationPageMetaModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
OrganizationPageMetaModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrganizationPageMetaModule });
OrganizationPageMetaModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrganizationPageMetaModule, providers: [
        {
            provide: PageMetaResolver,
            useExisting: OrganizationPageMetaResolver,
            multi: true,
        },
    ] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrganizationPageMetaModule, decorators: [{
            type: NgModule,
            args: [{
                    providers: [
                        {
                            provide: PageMetaResolver,
                            useExisting: OrganizationPageMetaResolver,
                            multi: true,
                        },
                    ],
                }]
        }] });
