// @spartacus/organization/esm2015/administration/components/shared/message/base-message.component.js
import { isPlatformBrowser } from '@angular/common';
import { Directive, HostBinding, Inject, PLATFORM_ID, } from '@angular/core';
import { GlobalMessageType } from '@spartacus/core';
import * as i0 from "@angular/core";
import * as i1 from "./message.model";
// eslint-disable-next-line @angular-eslint/directive-class-suffix
export class BaseMessageComponent {
    constructor(messageData, platformId) {
        this.messageData = messageData;
        this.platformId = platformId;
        this.terminated = false;
    }
    ngOnInit() {
        var _a;
        this.message = (_a = this.messageData.message) !== null && _a !== void 0 ? _a : {};
        this.messageTitle = this.messageData.messageTitle;
        this.type = this.resolveType();
        this.messageIcon = this.messageData.messageIcon;
        if (this.messageData.timeout) {
            this.handleAutoHide();
        }
    }
    close() {
        this.messageData.events.next({ close: true });
    }
    resolveType() {
        if (!this.messageData.type ||
            this.messageData.type === GlobalMessageType.MSG_TYPE_INFO) {
            return 'info';
        }
        if (this.messageData.type === GlobalMessageType.MSG_TYPE_ERROR) {
            return 'error';
        }
        if (this.messageData.type === GlobalMessageType.MSG_TYPE_WARNING) {
            return 'warning';
        }
    }
    handleAutoHide() {
        if (isPlatformBrowser(this.platformId)) {
            // we don't want to run this logic when doing SSR
            setTimeout(() => {
                this.close();
            }, this.messageData.timeout);
        }
    }
}
BaseMessageComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BaseMessageComponent, deps: [{ token: i1.MessageData }, { token: PLATFORM_ID }], target: i0.ɵɵFactoryTarget.Directive });
BaseMessageComponent.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.0.5", type: BaseMessageComponent, host: { properties: { "class": "this.type", "class.terminated": "this.terminated" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BaseMessageComponent, decorators: [{
            type: Directive
        }], ctorParameters: function () { return [{ type: i1.MessageData }, { type: undefined, decorators: [{
                    type: Inject,
                    args: [PLATFORM_ID]
                }] }]; }, propDecorators: { type: [{
                type: HostBinding,
                args: ['class']
            }], terminated: [{
                type: HostBinding,
                args: ['class.terminated']
            }] } });
