// @spartacus/organization/esm2015/administration/components/shared/message/message.component.js
import { ChangeDetectionStrategy, ChangeDetectorRef, Component, ViewChild, ViewContainerRef, } from '@angular/core';
import { Subscription } from 'rxjs';
import { filter } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "./services/message.service";
import * as i2 from "./services/message-render.service";
export class MessageComponent {
    constructor(messageService, renderService) {
        this.messageService = messageService;
        this.renderService = renderService;
        this.subscription = new Subscription();
    }
    ngAfterViewInit() {
        this.subscription.add(this.messageService.get().subscribe((msg) => {
            var _a;
            if (msg) {
                this.render(msg);
            }
            else {
                (_a = this.vcr) === null || _a === void 0 ? void 0 : _a.clear();
            }
        }));
    }
    render(msg) {
        const ref = this.vcr.createComponent(this.renderService.getComponent(msg), 0, this.renderService.getInjector(msg, this.vcr.injector));
        ref.injector.get(ChangeDetectorRef).markForCheck();
        this.subscription.add(msg.events
            .pipe(filter((event) => !!event.close))
            .subscribe(() => this.terminate(ref)));
    }
    /**
     * Terminates the message component in 2 steps. It starts to toggle the terminate
     * state of the component and shortly after destroys the component completely. The
     * termination state allows the CSS layer to play an animation before destroying.
     */
    terminate(ref) {
        ref.instance.terminated = true;
        ref.injector.get(ChangeDetectorRef).markForCheck();
        setTimeout(() => {
            ref.destroy();
        }, 500);
    }
    ngOnDestroy() {
        this.subscription.unsubscribe();
    }
}
MessageComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: MessageComponent, deps: [{ token: i1.MessageService }, { token: i2.MessageRenderService }], target: i0.ɵɵFactoryTarget.Component });
MessageComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: MessageComponent, selector: "cx-org-message", viewQueries: [{ propertyName: "vcr", first: true, predicate: ["vcr"], descendants: true, read: ViewContainerRef }], ngImport: i0, template: "<ng-container #vcr></ng-container>\n", changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: MessageComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'cx-org-message',
                    templateUrl: './message.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return [{ type: i1.MessageService }, { type: i2.MessageRenderService }]; }, propDecorators: { vcr: [{
                type: ViewChild,
                args: ['vcr', { read: ViewContainerRef }]
            }] } });
