// @spartacus/organization/esm2015/administration/components/permission/form/permission-form.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { NgSelectModule } from '@ng-select/ng-select';
import { I18nModule, UrlModule } from '@spartacus/core';
import { FormErrorsModule } from '@spartacus/storefront';
import { ItemActiveModule } from '../../shared/item-active.module';
import { FormModule } from '../../shared/form/form.module';
import { PermissionFormComponent } from './permission-form.component';
import * as i0 from "@angular/core";
export class PermissionFormModule {
}
PermissionFormModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PermissionFormModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
PermissionFormModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PermissionFormModule, declarations: [PermissionFormComponent], imports: [CommonModule,
        RouterModule,
        NgSelectModule,
        UrlModule,
        I18nModule,
        ReactiveFormsModule,
        FormErrorsModule,
        FormModule,
        ItemActiveModule], exports: [PermissionFormComponent] });
PermissionFormModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PermissionFormModule, imports: [[
            CommonModule,
            RouterModule,
            NgSelectModule,
            UrlModule,
            I18nModule,
            ReactiveFormsModule,
            FormErrorsModule,
            FormModule,
            ItemActiveModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PermissionFormModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        RouterModule,
                        NgSelectModule,
                        UrlModule,
                        I18nModule,
                        ReactiveFormsModule,
                        FormErrorsModule,
                        FormModule,
                        ItemActiveModule,
                    ],
                    declarations: [PermissionFormComponent],
                    exports: [PermissionFormComponent],
                }]
        }] });
