// @spartacus/organization/fesm2015/spartacus-organization-order-approval-root.js
import * as i0 from '@angular/core';
import { NgModule } from '@angular/core';
import * as i1 from '@angular/router';
import { RouterModule } from '@angular/router';
import { provideDefaultConfigFactory, provideDefaultConfig, AuthGuard } from '@spartacus/core';
import { CmsPageGuard, PageLayoutComponent } from '@spartacus/storefront';

const ORGANIZATION_ORDER_APPROVAL_FEATURE = 'organizationOrderApproval';

// TODO: Inline this factory when we start releasing Ivy compiled libraries
function defaultOrganizationOrderApprovalComponentsConfig() {
    const config = {
        featureModules: {
            [ORGANIZATION_ORDER_APPROVAL_FEATURE]: {
                cmsComponents: [
                    'OrderApprovalListComponent',
                    'OrderApprovalDetailTotalsComponent',
                    'OrderApprovalDetailApprovalDetailsComponent',
                    'OrderApprovalDetailShippingComponent',
                    'OrderApprovalDetailItemsComponent',
                    'OrderApprovalDetailFormComponent',
                    'AccountOrderDetailsApprovalDetailsComponent',
                ],
            },
        },
    };
    return config;
}
class OrderApprovalRootModule {
}
OrderApprovalRootModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderApprovalRootModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
OrderApprovalRootModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderApprovalRootModule, imports: [i1.RouterModule] });
OrderApprovalRootModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderApprovalRootModule, providers: [
        provideDefaultConfigFactory(defaultOrganizationOrderApprovalComponentsConfig),
        provideDefaultConfig({
            routing: {
                routes: {
                    orderApprovals: {
                        paths: ['my-account/approval-dashboard'],
                    },
                    orderApprovalDetails: {
                        paths: ['my-account/approval/:approvalCode'],
                        paramsMapping: { approvalCode: 'approvalCode' },
                    },
                },
            },
        }),
    ], imports: [[
            RouterModule.forChild([
                {
                    path: null,
                    canActivate: [AuthGuard, CmsPageGuard],
                    component: PageLayoutComponent,
                    data: { cxRoute: 'orderApprovalDetails' },
                },
            ]),
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderApprovalRootModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        RouterModule.forChild([
                            {
                                path: null,
                                canActivate: [AuthGuard, CmsPageGuard],
                                component: PageLayoutComponent,
                                data: { cxRoute: 'orderApprovalDetails' },
                            },
                        ]),
                    ],
                    providers: [
                        provideDefaultConfigFactory(defaultOrganizationOrderApprovalComponentsConfig),
                        provideDefaultConfig({
                            routing: {
                                routes: {
                                    orderApprovals: {
                                        paths: ['my-account/approval-dashboard'],
                                    },
                                    orderApprovalDetails: {
                                        paths: ['my-account/approval/:approvalCode'],
                                        paramsMapping: { approvalCode: 'approvalCode' },
                                    },
                                },
                            },
                        }),
                    ],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { ORGANIZATION_ORDER_APPROVAL_FEATURE, OrderApprovalRootModule, defaultOrganizationOrderApprovalComponentsConfig };
