// @spartacus/organization/esm2015/administration/components/shared/detail/delete-item-action/delete-item.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { I18nModule } from '@spartacus/core';
import { ConfirmationMessageModule } from '../../message/confirmation/confirmation-message.module';
import { MessageModule } from '../../message/message.module';
import { DeleteItemComponent } from './delete-item.component';
import * as i0 from "@angular/core";
export class DeleteItemModule {
}
DeleteItemModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: DeleteItemModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
DeleteItemModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: DeleteItemModule, declarations: [DeleteItemComponent], imports: [CommonModule, I18nModule, MessageModule, ConfirmationMessageModule], exports: [DeleteItemComponent] });
DeleteItemModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: DeleteItemModule, imports: [[CommonModule, I18nModule, MessageModule, ConfirmationMessageModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: DeleteItemModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, I18nModule, MessageModule, ConfirmationMessageModule],
                    declarations: [DeleteItemComponent],
                    exports: [DeleteItemComponent],
                }]
        }] });
