// @spartacus/organization/esm2015/administration/components/user/services/current-user.service.js
import { Injectable } from '@angular/core';
import { ROUTE_PARAMS } from '@spartacus/organization/administration/root';
import { map } from 'rxjs/operators';
import { CurrentItemService } from '../../shared/current-item.service';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
import * as i2 from "@spartacus/organization/administration/core";
export class CurrentUserService extends CurrentItemService {
    constructor(routingService, b2bUserService) {
        super(routingService);
        this.routingService = routingService;
        this.b2bUserService = b2bUserService;
        this.name$ = this.item$.pipe(map((user) => user.name));
    }
    getParamKey() {
        return ROUTE_PARAMS.userCode;
    }
    getItem(code) {
        return this.b2bUserService.get(code);
    }
    getError(code) {
        return this.b2bUserService.getErrorState(code);
    }
}
CurrentUserService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CurrentUserService, deps: [{ token: i1.RoutingService }, { token: i2.B2BUserService }], target: i0.ɵɵFactoryTarget.Injectable });
CurrentUserService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CurrentUserService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CurrentUserService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.RoutingService }, { type: i2.B2BUserService }]; } });
