// @spartacus/organization/esm2015/administration/components/user/services/user-item.service.js
import { Injectable } from '@angular/core';
import { ItemService } from '../../shared/item.service';
import * as i0 from "@angular/core";
import * as i1 from "./current-user.service";
import * as i2 from "@spartacus/core";
import * as i3 from "../form/user-form.service";
import * as i4 from "@spartacus/organization/administration/core";
export class UserItemService extends ItemService {
    constructor(currentItemService, routingService, formService, userService) {
        super(currentItemService, routingService, formService);
        this.currentItemService = currentItemService;
        this.routingService = routingService;
        this.formService = formService;
        this.userService = userService;
    }
    load(code) {
        this.userService.load(code);
        return this.userService.get(code);
    }
    update(code, value) {
        delete value.approvers;
        this.userService.update(code, value);
        return this.userService.getLoadingStatus(code);
    }
    create(value) {
        this.userService.create(value);
        return this.userService.getLoadingStatus(null);
    }
    getDetailsRoute() {
        return 'orgUserDetails';
    }
    // @override to avoid errors while creation
    launchDetails(item) {
        if (item.customerId !== null) {
            super.launchDetails(item);
        }
    }
}
UserItemService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserItemService, deps: [{ token: i1.CurrentUserService }, { token: i2.RoutingService }, { token: i3.UserFormService }, { token: i4.B2BUserService }], target: i0.ɵɵFactoryTarget.Injectable });
UserItemService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserItemService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserItemService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.CurrentUserService }, { type: i2.RoutingService }, { type: i3.UserFormService }, { type: i4.B2BUserService }]; } });
