// @spartacus/organization/esm2015/administration/components/unit/links/addresses/services/current-unit-address.service.js
import { Injectable } from '@angular/core';
import { ROUTE_PARAMS } from '@spartacus/organization/administration/root';
import { of } from 'rxjs';
import { filter, switchMap } from 'rxjs/operators';
import { CurrentItemService } from '../../../../shared/current-item.service';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
import * as i2 from "@spartacus/organization/administration/core";
export class CurrentUnitAddressService extends CurrentItemService {
    constructor(routingService, unitService) {
        super(routingService);
        this.routingService = routingService;
        this.unitService = unitService;
        // override item$ as we need to use the unit code as well
        this.item$ = this.b2bUnit$.pipe(filter((unitUid) => Boolean(unitUid)), switchMap((unitUid) => this.key$.pipe(switchMap((code) => this.getItem(unitUid, code)))));
    }
    getDetailsRoute() {
        return 'orgUnitAddressDetails';
    }
    getParamKey() {
        return ROUTE_PARAMS.addressCode;
    }
    getItem(unitUid, addressId) {
        return addressId
            ? this.unitService.getAddress(unitUid, addressId)
            : of(null);
    }
    getError(code) {
        return this.unitService.getErrorState(code);
    }
}
CurrentUnitAddressService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CurrentUnitAddressService, deps: [{ token: i1.RoutingService }, { token: i2.OrgUnitService }], target: i0.ɵɵFactoryTarget.Injectable });
CurrentUnitAddressService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CurrentUnitAddressService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CurrentUnitAddressService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.RoutingService }, { type: i2.OrgUnitService }]; } });
