// @spartacus/organization/esm2015/administration/components/budget/details-cell/budget-details-cell.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { I18nModule, UrlModule } from '@spartacus/core';
import { PopoverModule } from '@spartacus/storefront';
import { BudgetDetailsCellComponent } from './budget-details-cell.component';
import * as i0 from "@angular/core";
export class BudgetDetailsCellModule {
}
BudgetDetailsCellModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BudgetDetailsCellModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
BudgetDetailsCellModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BudgetDetailsCellModule, declarations: [BudgetDetailsCellComponent], imports: [CommonModule, PopoverModule, RouterModule, I18nModule, UrlModule], exports: [BudgetDetailsCellComponent] });
BudgetDetailsCellModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BudgetDetailsCellModule, imports: [[CommonModule, PopoverModule, RouterModule, I18nModule, UrlModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BudgetDetailsCellModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, PopoverModule, RouterModule, I18nModule, UrlModule],
                    declarations: [BudgetDetailsCellComponent],
                    exports: [BudgetDetailsCellComponent],
                }]
        }] });
