// @spartacus/organization/esm2015/administration/components/user/user-groups/assigned/user-assigned-user-group-list.service.js
import { Injectable } from '@angular/core';
import { filter, map } from 'rxjs/operators';
import { OrganizationTableType } from '../../../shared/organization.model';
import { UserUserGroupListService } from '../user-user-group-list.service';
import * as i0 from "@angular/core";
export class UserAssignedUserGroupListService extends UserUserGroupListService {
    constructor() {
        super(...arguments);
        this.tableType = OrganizationTableType.USER_ASSIGNED_USER_GROUPS;
    }
    load(pagination, code) {
        return super.load(pagination, code).pipe(filter((list) => Boolean(list)), map((userGroups) => this.filterSelected(userGroups)));
    }
}
UserAssignedUserGroupListService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserAssignedUserGroupListService, deps: null, target: i0.ɵɵFactoryTarget.Injectable });
UserAssignedUserGroupListService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserAssignedUserGroupListService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserAssignedUserGroupListService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }] });
