// @spartacus/organization/esm2015/administration/components/unit/details/unit-details.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { I18nModule, UrlModule } from '@spartacus/core';
import { KeyboardFocusModule } from '@spartacus/storefront';
import { CardModule } from '../../shared/card/card.module';
import { DisableInfoModule } from '../../shared/detail/disable-info/disable-info.module';
import { ToggleStatusModule } from '../../shared/detail/toggle-status-action/toggle-status.module';
import { ItemExistsModule } from '../../shared/item-exists.module';
import { UnitDetailsComponent } from './unit-details.component';
import * as i0 from "@angular/core";
export class UnitDetailsModule {
}
UnitDetailsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitDetailsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
UnitDetailsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitDetailsModule, declarations: [UnitDetailsComponent], imports: [CommonModule,
        CardModule,
        RouterModule,
        UrlModule,
        I18nModule,
        ToggleStatusModule,
        ItemExistsModule,
        KeyboardFocusModule,
        DisableInfoModule], exports: [UnitDetailsComponent] });
UnitDetailsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitDetailsModule, imports: [[
            CommonModule,
            CardModule,
            RouterModule,
            UrlModule,
            I18nModule,
            ToggleStatusModule,
            ItemExistsModule,
            KeyboardFocusModule,
            DisableInfoModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitDetailsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        CardModule,
                        RouterModule,
                        UrlModule,
                        I18nModule,
                        ToggleStatusModule,
                        ItemExistsModule,
                        KeyboardFocusModule,
                        DisableInfoModule,
                    ],
                    declarations: [UnitDetailsComponent],
                    exports: [UnitDetailsComponent],
                }]
        }] });
