// @spartacus/organization/esm2015/administration/components/shared/message/services/message-render.service.js
import { Injectable, Injector, } from '@angular/core';
import { MessageData } from '../message.model';
import { NotificationMessageComponent } from '../notification/notification-message.component';
import * as i0 from "@angular/core";
export class MessageRenderService {
    constructor(componentFactoryResolver) {
        this.componentFactoryResolver = componentFactoryResolver;
    }
    getComponent(msg) {
        return this.componentFactoryResolver.resolveComponentFactory(msg.component || NotificationMessageComponent);
    }
    getInjector(componentData, parent) {
        return Injector.create({
            providers: [
                {
                    provide: MessageData,
                    useValue: componentData,
                },
            ],
            parent,
        });
    }
}
MessageRenderService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: MessageRenderService, deps: [{ token: i0.ComponentFactoryResolver }], target: i0.ɵɵFactoryTarget.Injectable });
MessageRenderService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: MessageRenderService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: MessageRenderService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i0.ComponentFactoryResolver }]; } });
