// @spartacus/organization/esm2015/administration/components/shared/message/services/message.service.js
import { Injectable } from '@angular/core';
import { GlobalMessageType } from '@spartacus/core';
import { ReplaySubject, Subject } from 'rxjs';
import * as i0 from "@angular/core";
const DEFAULT_INFO_TIMEOUT = 3000;
export class MessageService {
    constructor() {
        this.data$ = new ReplaySubject();
    }
    get() {
        return this.data$;
    }
    add(message) {
        message = Object.assign(Object.assign({}, this.getDefaultMessage(message)), message);
        message.events = new Subject();
        this.data$.next(message);
        return message.events;
    }
    close(message) {
        message.next({ close: true });
    }
    /**
     * Sets the message type to INFO, and adds a default timeout
     * for info messages.
     */
    getDefaultMessage(message) {
        const defaultMessage = {
            type: GlobalMessageType.MSG_TYPE_INFO,
        };
        if (!message.type ||
            (message.type === GlobalMessageType.MSG_TYPE_INFO && !message.timeout)) {
            defaultMessage.timeout = DEFAULT_INFO_TIMEOUT;
        }
        return defaultMessage;
    }
    clear() {
        this.data$.next();
    }
}
MessageService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: MessageService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
MessageService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: MessageService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: MessageService, decorators: [{
            type: Injectable
        }] });
