// @spartacus/organization/esm2015/administration/components/unit/list/unit-list.component.js
import { ChangeDetectionStrategy, Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../services/unit-tree.service";
import * as i2 from "../../shared/list/list.component";
import * as i3 from "@spartacus/core";
export class UnitListComponent {
    constructor(unitTreeService) {
        this.unitTreeService = unitTreeService;
    }
    expandAll() {
        this.unitTreeService.expandAll();
    }
    collapseAll() {
        this.unitTreeService.collapseAll();
    }
}
UnitListComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitListComponent, deps: [{ token: i1.UnitTreeService }], target: i0.ɵɵFactoryTarget.Component });
UnitListComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: UnitListComponent, selector: "cx-org-unit-list", ngImport: i0, template: "<cx-org-list>\n  <ng-container actions>\n    <button class=\"link\" (click)=\"expandAll()\">\n      {{ 'orgUnit.tree.expandAll' | cxTranslate }}\n    </button>\n    <button class=\"link\" (click)=\"collapseAll()\">\n      {{ 'orgUnit.tree.collapseAll' | cxTranslate }}\n    </button>\n  </ng-container>\n</cx-org-list>\n", components: [{ type: i2.ListComponent, selector: "cx-org-list", inputs: ["key"] }], pipes: { "cxTranslate": i3.TranslatePipe }, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitListComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'cx-org-unit-list',
                    templateUrl: './unit-list.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return [{ type: i1.UnitTreeService }]; } });
