// @spartacus/organization/esm2015/administration/core/store/reducers/b2b-user.reducer.js
import { B2BUserActions, OrgUnitActions, UserGroupActions, } from '../actions/index';
export const b2bUserInitialState = undefined;
export const b2bUsersInitialState = undefined;
export function b2bUserEntitiesReducer(state = b2bUserInitialState, action) {
    var _a, _b, _c;
    switch (action.type) {
        case B2BUserActions.LOAD_B2B_USER_SUCCESS:
        case B2BUserActions.CREATE_B2B_USER_SUCCESS:
        case B2BUserActions.UPDATE_B2B_USER_SUCCESS:
            return action.payload;
        case OrgUnitActions.ASSIGN_ROLE_SUCCESS:
        case OrgUnitActions.ASSIGN_APPROVER_SUCCESS:
            return Object.assign(Object.assign({}, state), { selected: (_a = action.payload) === null || _a === void 0 ? void 0 : _a.selected, roles: [...((state === null || state === void 0 ? void 0 : state.roles) || []), (_b = action.payload) === null || _b === void 0 ? void 0 : _b.roleId] });
        case OrgUnitActions.UNASSIGN_ROLE_SUCCESS:
        case OrgUnitActions.UNASSIGN_APPROVER_SUCCESS:
            return Object.assign(Object.assign({}, state), { selected: (_c = action.payload) === null || _c === void 0 ? void 0 : _c.selected, roles: [...((state === null || state === void 0 ? void 0 : state.roles) || [])].filter((role) => { var _a; return role !== ((_a = action.payload) === null || _a === void 0 ? void 0 : _a.roleId); }) });
        case B2BUserActions.ASSIGN_B2B_USER_APPROVER_SUCCESS:
        case B2BUserActions.UNASSIGN_B2B_USER_APPROVER_SUCCESS:
        case UserGroupActions.USER_GROUP_ASSIGN_MEMBER_SUCCESS:
        case UserGroupActions.USER_GROUP_UNASSIGN_MEMBER_SUCCESS:
        case UserGroupActions.USER_GROUP_UNASSIGN_ALL_MEMBERS_SUCCESS:
            return Object.assign(Object.assign({}, state), action.payload);
    }
    return state;
}
export function userListReducer(state = b2bUsersInitialState, action) {
    switch (action.type) {
        case B2BUserActions.LOAD_B2B_USERS_SUCCESS:
            return action.payload.page;
    }
    return state;
}
export function b2bUserApproverListReducer(state = b2bUsersInitialState, action) {
    switch (action.type) {
        case B2BUserActions.LOAD_B2B_USER_APPROVERS_SUCCESS:
            return action.payload.page;
    }
    return state;
}
export function b2bUserPermissionListReducer(state = b2bUsersInitialState, action) {
    switch (action.type) {
        case B2BUserActions.LOAD_B2B_USER_PERMISSIONS_SUCCESS:
            return action.payload.page;
    }
    return state;
}
export function b2bUserUserGroupListReducer(state = b2bUsersInitialState, action) {
    switch (action.type) {
        case B2BUserActions.LOAD_B2B_USER_USER_GROUPS_SUCCESS:
            return action.payload.page;
    }
    return state;
}
