// @spartacus/organization/esm2015/administration/core/store/reducers/org-unit.reducer.js
import { OrgUnitActions } from '../actions/index';
export const orgUnitInitialState = undefined;
export const orgUnitsInitialState = undefined;
export function orgUnitEntitiesReducer(state = orgUnitInitialState, action) {
    switch (action.type) {
        case OrgUnitActions.LOAD_ORG_UNIT_SUCCESS:
        case OrgUnitActions.CREATE_ORG_UNIT_SUCCESS:
            return action.payload;
        case OrgUnitActions.UPDATE_ORG_UNIT_SUCCESS:
            return Object.assign(Object.assign({}, state), action.payload);
    }
    return state;
}
export function orgUnitListReducer(state = orgUnitsInitialState, action) {
    switch (action.type) {
    }
    return state;
}
export function orgUnitUserListReducer(state = orgUnitsInitialState, action) {
    switch (action.type) {
        case OrgUnitActions.LOAD_ASSIGNED_USERS_SUCCESS:
            return action.payload.page;
    }
    return state;
}
export function orgUnitAddressListReducer(state = orgUnitsInitialState, action) {
    switch (action.type) {
        case OrgUnitActions.LOAD_ADDRESSES_SUCCESS:
            return action.payload.page;
    }
    return state;
}
