// @spartacus/organization/esm2015/administration/core/store/reducers/user-group.reducer.js
import * as UserGroupActions from '../actions/user-group.action';
import { B2BUserActions } from '../actions/index';
export const userGroupInitialState = undefined;
export const userGroupsInitialState = undefined;
export function userGroupEntitiesReducer(state = userGroupInitialState, action) {
    switch (action.type) {
        case UserGroupActions.LOAD_USER_GROUP_SUCCESS:
        case UserGroupActions.CREATE_USER_GROUP_SUCCESS:
        case UserGroupActions.UPDATE_USER_GROUP_SUCCESS:
            return action.payload;
        case B2BUserActions.ASSIGN_B2B_USER_USER_GROUP_SUCCESS:
        case B2BUserActions.UNASSIGN_B2B_USER_USER_GROUP_SUCCESS:
            return Object.assign(Object.assign({}, state), action.payload);
    }
    return state;
}
export function userGroupsListReducer(state = userGroupsInitialState, action) {
    switch (action.type) {
        case UserGroupActions.LOAD_USER_GROUPS_SUCCESS:
            return action.payload.page;
    }
    return state;
}
export function userGroupAvailableOrderApprovalPermissionsListReducer(state = userGroupsInitialState, action) {
    switch (action.type) {
        case UserGroupActions.LOAD_USER_GROUP_PERMISSIONS_SUCCESS:
            return action.payload.page;
    }
    return state;
}
export function userGroupAvailablOrgCustomersListReducer(state = userGroupsInitialState, action) {
    switch (action.type) {
        case UserGroupActions.LOAD_USER_GROUP_AVAILABLE_CUSTOMERS_SUCCESS:
            return action.payload.page;
    }
    return state;
}
