// @spartacus/organization/esm2015/administration/components/shared/sub-list/sub-list.service.js
import { Injectable } from '@angular/core';
import { TableLayout, } from '@spartacus/storefront';
import { ListService } from '../list/list.service';
import { of } from 'rxjs';
import * as i0 from "@angular/core";
export class SubListService extends ListService {
    constructor() {
        super(...arguments);
        /**
         * The default table structure for sub lists is only showing tables with vertical layout.
         */
        this.defaultTableStructure = {
            options: { layout: TableLayout.VERTICAL },
        };
        /**
         * @override This sub list will show 3 items.
         */
        this.ghostData = { values: new Array(3) };
    }
    // TODO: abstract
    assign(_key, ..._args) {
        return of();
    }
    unassign(_key, ..._args) {
        return of();
    }
    /**
     * As we can't filter with the backend API, we do this client side.
     */
    filterSelected(list) {
        if (!list) {
            return list;
        }
        const { pagination, sorts, values } = list;
        return {
            pagination,
            sorts,
            values: values.filter((value) => value.selected),
        };
    }
}
SubListService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SubListService, deps: null, target: i0.ɵɵFactoryTarget.Injectable });
SubListService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SubListService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SubListService, decorators: [{
            type: Injectable
        }] });
