// @spartacus/organization/esm2015/administration/components/shared/sub-list/sub-list.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { I18nModule } from '@spartacus/core';
import { KeyboardFocusModule, PaginationModule, TableModule, } from '@spartacus/storefront';
import { CardModule } from '../card/card.module';
import { MessageModule } from '../message/message.module';
import { AssignCellComponent } from './assign-cell.component';
import { SubListComponent } from './sub-list.component';
import * as i0 from "@angular/core";
export class SubListModule {
}
SubListModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SubListModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
SubListModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SubListModule, declarations: [SubListComponent, AssignCellComponent], imports: [CommonModule,
        I18nModule,
        CardModule,
        TableModule,
        PaginationModule,
        MessageModule,
        KeyboardFocusModule], exports: [SubListComponent] });
SubListModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SubListModule, imports: [[
            CommonModule,
            I18nModule,
            CardModule,
            TableModule,
            PaginationModule,
            MessageModule,
            KeyboardFocusModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: SubListModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        I18nModule,
                        CardModule,
                        TableModule,
                        PaginationModule,
                        MessageModule,
                        KeyboardFocusModule,
                    ],
                    declarations: [SubListComponent, AssignCellComponent],
                    exports: [SubListComponent],
                }]
        }] });
