// @spartacus/organization/esm2015/administration/components/shared/table/cell.component.js
import { __rest } from "tslib";
import { ChangeDetectionStrategy, Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/storefront";
import * as i2 from "@angular/common";
import * as i3 from "@angular/router";
import * as i4 from "@spartacus/core";
export class CellComponent {
    constructor(outlet) {
        this.outlet = outlet;
    }
    get tabIndex() {
        return -1;
    }
    get model() {
        return this.outlet.context;
    }
    get property() {
        var _a, _b, _c;
        return (_a = this.model) === null || _a === void 0 ? void 0 : _a[(_c = (_b = this.outlet) === null || _b === void 0 ? void 0 : _b.context) === null || _c === void 0 ? void 0 : _c._field];
    }
    /**
     * Indicates wether the cell is linkable.
     *
     * If the cells is linkable, an anchor link is created to the detailed route
     * of the given `_type`.
     *
     * Defaults to `true`.
     */
    get linkable() {
        var _a;
        return this.property !== undefined && ((_a = this.cellOptions.linkable) !== null && _a !== void 0 ? _a : true);
    }
    /**
     * Helper method to access the cell options.
     */
    get cellOptions() {
        var _a, _b, _c, _d, _e;
        return ((_e = (_c = (_b = (_a = this.outlet.context) === null || _a === void 0 ? void 0 : _a._options) === null || _b === void 0 ? void 0 : _b.cells) === null || _c === void 0 ? void 0 : _c[(_d = this.outlet.context) === null || _d === void 0 ? void 0 : _d._field]) !== null && _e !== void 0 ? _e : {});
    }
    /**
     * Generates the configurable route to the detail page of the given context item.
     */
    get route() {
        return this.outlet.context._type + 'Details';
    }
    get routeModel() {
        return this.outlet.context;
    }
    get type() {
        return this.model._type;
    }
    /**
     * Indicates whether the item is loaded.
     */
    get hasItem() {
        return !!this.item && Object.keys(this.item).length > 0;
    }
    get item() {
        if (!this.outlet.context) {
            return null;
        }
        const _a = this.outlet.context, { _field, _options, _type, _i18nRoot } = _a, all = __rest(_a, ["_field", "_options", "_type", "_i18nRoot"]);
        return all;
    }
}
CellComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CellComponent, deps: [{ token: i1.OutletContextData }], target: i0.ɵɵFactoryTarget.Component });
CellComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: CellComponent, selector: "cx-org-cell", ngImport: i0, template: "<a\n  *ngIf=\"linkable; else text\"\n  [routerLink]=\"{ cxRoute: route, params: routeModel } | cxUrl\"\n  [tabIndex]=\"tabIndex\"\n>\n  <ng-container *ngTemplateOutlet=\"text\"></ng-container>\n</a>\n\n<ng-template #text>\n  <span class=\"text\" title=\"{{ property }}\">{{ property }}</span>\n</ng-template>\n", directives: [{ type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i3.RouterLinkWithHref, selector: "a[routerLink],area[routerLink]", inputs: ["routerLink", "target", "queryParams", "fragment", "queryParamsHandling", "preserveFragment", "skipLocationChange", "replaceUrl", "state", "relativeTo"] }, { type: i2.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet"] }], pipes: { "cxUrl": i4.UrlPipe }, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CellComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'cx-org-cell',
                    templateUrl: './cell.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return [{ type: i1.OutletContextData }]; } });
