// @spartacus/organization/esm2015/order-approval/components/list/order-approval-list.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { AuthGuard, ConfigModule, I18nModule, UrlModule, } from '@spartacus/core';
import { ListNavigationModule } from '@spartacus/storefront';
import { ApproverGuard } from '../../core/guards/approver.guard';
import { OrderApprovalListComponent } from './order-approval-list.component';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
export class OrderApprovalListModule {
}
OrderApprovalListModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderApprovalListModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
OrderApprovalListModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderApprovalListModule, declarations: [OrderApprovalListComponent], imports: [CommonModule, i1.ConfigModule, UrlModule,
        RouterModule,
        ListNavigationModule,
        I18nModule], exports: [OrderApprovalListComponent] });
OrderApprovalListModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderApprovalListModule, imports: [[
            CommonModule,
            ConfigModule.withConfig({
                cmsComponents: {
                    OrderApprovalListComponent: {
                        component: OrderApprovalListComponent,
                        guards: [AuthGuard, ApproverGuard],
                    },
                },
            }),
            UrlModule,
            RouterModule,
            ListNavigationModule,
            I18nModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderApprovalListModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        ConfigModule.withConfig({
                            cmsComponents: {
                                OrderApprovalListComponent: {
                                    component: OrderApprovalListComponent,
                                    guards: [AuthGuard, ApproverGuard],
                                },
                            },
                        }),
                        UrlModule,
                        RouterModule,
                        ListNavigationModule,
                        I18nModule,
                    ],
                    declarations: [OrderApprovalListComponent],
                    exports: [OrderApprovalListComponent],
                }]
        }] });
