// @spartacus/organization/esm2015/administration/components/cost-center/form/cost-center-form.service.js
import { Injectable } from '@angular/core';
import { FormControl, FormGroup, Validators } from '@angular/forms';
import { CustomFormValidators } from '@spartacus/storefront';
import { FormService } from '../../shared/form/form.service';
import * as i0 from "@angular/core";
export class CostCenterFormService extends FormService {
    build() {
        const form = new FormGroup({});
        form.setControl('code', new FormControl('', [
            Validators.required,
            CustomFormValidators.noSpecialCharacters,
        ]));
        form.setControl('name', new FormControl('', Validators.required));
        form.setControl('currency', new FormGroup({
            isocode: new FormControl(undefined, Validators.required),
        }));
        form.setControl('unit', new FormGroup({
            uid: new FormControl(undefined, Validators.required),
        }));
        this.form = form;
    }
}
CostCenterFormService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CostCenterFormService, deps: null, target: i0.ɵɵFactoryTarget.Injectable });
CostCenterFormService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CostCenterFormService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CostCenterFormService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }] });
