// @spartacus/organization/esm2015/administration/components/cost-center/form/cost-center-form.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { NgSelectModule } from '@ng-select/ng-select';
import { CurrencyService, I18nModule, UrlModule } from '@spartacus/core';
import { OrgUnitService } from '@spartacus/organization/administration/core';
import { FormErrorsModule } from '@spartacus/storefront';
import { ItemActiveModule } from '../../shared/item-active.module';
import { FormModule } from '../../shared/form/form.module';
import { CostCenterFormComponent } from './cost-center-form.component';
import * as i0 from "@angular/core";
export class CostCenterFormModule {
}
CostCenterFormModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CostCenterFormModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CostCenterFormModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CostCenterFormModule, declarations: [CostCenterFormComponent], imports: [CommonModule,
        RouterModule,
        FormModule,
        NgSelectModule,
        UrlModule,
        I18nModule,
        ReactiveFormsModule,
        FormErrorsModule,
        ItemActiveModule], exports: [CostCenterFormComponent] });
CostCenterFormModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CostCenterFormModule, providers: [CurrencyService, OrgUnitService], imports: [[
            CommonModule,
            RouterModule,
            FormModule,
            NgSelectModule,
            UrlModule,
            I18nModule,
            ReactiveFormsModule,
            FormErrorsModule,
            ItemActiveModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CostCenterFormModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        RouterModule,
                        FormModule,
                        NgSelectModule,
                        UrlModule,
                        I18nModule,
                        ReactiveFormsModule,
                        FormErrorsModule,
                        ItemActiveModule,
                    ],
                    declarations: [CostCenterFormComponent],
                    exports: [CostCenterFormComponent],
                    providers: [CurrencyService, OrgUnitService],
                }]
        }] });
