// @spartacus/organization/esm2015/administration/components/unit/links/users/create/unit-user-create.component.js
import { ChangeDetectionStrategy, Component } from '@angular/core';
import { UserItemService } from '../../../../user/services/user-item.service';
import { UnitUserItemService } from './unit-user-item.service';
import * as i0 from "@angular/core";
import * as i1 from "../../../services/current-unit.service";
import * as i2 from "../../../../user/form/user-form.component";
import * as i3 from "@angular/common";
export class UnitUserCreateComponent {
    constructor(unitService) {
        this.unitService = unitService;
        this.unitKey$ = this.unitService.key$;
    }
}
UnitUserCreateComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitUserCreateComponent, deps: [{ token: i1.CurrentUnitService }], target: i0.ɵɵFactoryTarget.Component });
UnitUserCreateComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: UnitUserCreateComponent, selector: "cx-org-unit-user-create", host: { classAttribute: "content-wrapper" }, providers: [
        // we provide a specific version of the `UnitItemService` to
        // let the form component work with child units.
        {
            provide: UserItemService,
            useExisting: UnitUserItemService,
        },
    ], ngImport: i0, template: "<cx-org-user-form\n  [unitKey]=\"unitKey$ | async\"\n  i18nRoot=\"orgUnit.users\"\n></cx-org-user-form>\n", components: [{ type: i2.UserFormComponent, selector: "cx-org-user-form", inputs: ["unitKey"] }], pipes: { "async": i3.AsyncPipe }, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitUserCreateComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'cx-org-unit-user-create',
                    templateUrl: './unit-user-create.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    host: { class: 'content-wrapper' },
                    providers: [
                        // we provide a specific version of the `UnitItemService` to
                        // let the form component work with child units.
                        {
                            provide: UserItemService,
                            useExisting: UnitUserItemService,
                        },
                    ],
                }]
        }], ctorParameters: function () { return [{ type: i1.CurrentUnitService }]; } });
