// @spartacus/organization/esm2015/administration/components/unit/links/users/create/unit-user-item.service.js
import { Injectable } from '@angular/core';
import { UserItemService } from '../../../../user/services/user-item.service';
import * as i0 from "@angular/core";
export class UnitUserItemService extends UserItemService {
    save(form, key) {
        var _a;
        // we enable the orgUnit temporarily so that the underlying
        // save method can read the complete form.value.
        (_a = form.get('orgUnit')) === null || _a === void 0 ? void 0 : _a.enable();
        return super.save(form, key);
    }
    /**
     * @override
     * Returns 'unitDetails'
     */
    getDetailsRoute() {
        return 'orgUnitUserList';
    }
    buildRouteParams(item) {
        return { uid: item.orgUnit.uid };
    }
    // @override to default method
    launchDetails(item) {
        const cxRoute = this.getDetailsRoute();
        const params = this.buildRouteParams(item);
        if (cxRoute && item && Object.keys(item).length > 0) {
            this.routingService.go({ cxRoute, params });
        }
    }
}
UnitUserItemService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitUserItemService, deps: null, target: i0.ɵɵFactoryTarget.Injectable });
UnitUserItemService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitUserItemService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitUserItemService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }] });
