// @spartacus/organization/esm2015/administration/components/cost-center/services/cost-center-list.service.js
import { Injectable } from '@angular/core';
import { filter, map } from 'rxjs/operators';
import { ListService } from '../../shared/list/list.service';
import { OrganizationTableType } from '../../shared/organization.model';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/storefront";
import * as i2 from "@spartacus/organization/administration/core";
/**
 * Service to populate Cost Center data to `Table` data. The cost center
 * data is driven by the table configuration, using the `OrganizationTables.COST_CENTER`.
 */
export class CostCenterListService extends ListService {
    constructor(tableService, costCenterService) {
        super(tableService);
        this.tableService = tableService;
        this.costCenterService = costCenterService;
        this.tableType = OrganizationTableType.COST_CENTER;
    }
    load(pagination) {
        return this.costCenterService.getList(pagination).pipe(filter((list) => Boolean(list)), map((raw) => this.convertCostCenters(raw)));
    }
    /**
     * Populates the cost center data to a convenient table data model, so that we
     * can skip specific conversion in the view logic where possible.
     */
    convertCostCenters({ pagination, sorts, values, }) {
        const costCenterModels = {
            pagination,
            sorts,
            values: values.map((value) => {
                var _a;
                return (Object.assign(Object.assign({}, value), { currency: (_a = value.currency) === null || _a === void 0 ? void 0 : _a.isocode }));
            }),
        };
        return costCenterModels;
    }
}
CostCenterListService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CostCenterListService, deps: [{ token: i1.TableService }, { token: i2.CostCenterService }], target: i0.ɵɵFactoryTarget.Injectable });
CostCenterListService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CostCenterListService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CostCenterListService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.TableService }, { type: i2.CostCenterService }]; } });
