// @spartacus/organization/esm2015/administration/components/cost-center/services/cost-center-item.service.js
import { Injectable } from '@angular/core';
import { ItemService } from '../../shared/item.service';
import * as i0 from "@angular/core";
import * as i1 from "./current-cost-center.service";
import * as i2 from "@spartacus/core";
import * as i3 from "../form/cost-center-form.service";
import * as i4 from "@spartacus/organization/administration/core";
export class CostCenterItemService extends ItemService {
    constructor(currentItemService, routingService, formService, costCenterService) {
        super(currentItemService, routingService, formService);
        this.currentItemService = currentItemService;
        this.routingService = routingService;
        this.formService = formService;
        this.costCenterService = costCenterService;
    }
    load(code) {
        this.costCenterService.load(code);
        return this.costCenterService.get(code);
    }
    update(code, value) {
        this.costCenterService.update(code, value);
        return this.costCenterService.getLoadingStatus(value.code);
    }
    create(value) {
        this.costCenterService.create(value);
        return this.costCenterService.getLoadingStatus(value.code);
    }
    getDetailsRoute() {
        return 'orgCostCenterDetails';
    }
}
CostCenterItemService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CostCenterItemService, deps: [{ token: i1.CurrentCostCenterService }, { token: i2.RoutingService }, { token: i3.CostCenterFormService }, { token: i4.CostCenterService }], target: i0.ɵɵFactoryTarget.Injectable });
CostCenterItemService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CostCenterItemService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CostCenterItemService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.CurrentCostCenterService }, { type: i2.RoutingService }, { type: i3.CostCenterFormService }, { type: i4.CostCenterService }]; } });
