// @spartacus/organization/esm2015/administration/components/cost-center/services/current-cost-center.service.js
import { Injectable } from '@angular/core';
import { ROUTE_PARAMS } from '@spartacus/organization/administration/root';
import { CurrentItemService } from '../../shared/current-item.service';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
import * as i2 from "@spartacus/organization/administration/core";
export class CurrentCostCenterService extends CurrentItemService {
    constructor(routingService, costCenterService) {
        super(routingService);
        this.routingService = routingService;
        this.costCenterService = costCenterService;
    }
    getParamKey() {
        return ROUTE_PARAMS.costCenterCode;
    }
    getItem(code) {
        return this.costCenterService.get(code);
    }
    getError(code) {
        return this.costCenterService.getErrorState(code);
    }
}
CurrentCostCenterService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CurrentCostCenterService, deps: [{ token: i1.RoutingService }, { token: i2.CostCenterService }], target: i0.ɵɵFactoryTarget.Injectable });
CurrentCostCenterService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CurrentCostCenterService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CurrentCostCenterService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.RoutingService }, { type: i2.CostCenterService }]; } });
