// @spartacus/organization/esm2015/administration/components/user/details-cell/user-details-cell.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { I18nModule, UrlModule } from '@spartacus/core';
import { PopoverModule } from '@spartacus/storefront';
import { UserDetailsCellComponent } from './user-details-cell.component';
import * as i0 from "@angular/core";
export class UserDetailsCellModule {
}
UserDetailsCellModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserDetailsCellModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
UserDetailsCellModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserDetailsCellModule, declarations: [UserDetailsCellComponent], imports: [CommonModule, PopoverModule, RouterModule, I18nModule, UrlModule], exports: [UserDetailsCellComponent] });
UserDetailsCellModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserDetailsCellModule, imports: [[CommonModule, PopoverModule, RouterModule, I18nModule, UrlModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserDetailsCellModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, PopoverModule, RouterModule, I18nModule, UrlModule],
                    declarations: [UserDetailsCellComponent],
                    exports: [UserDetailsCellComponent],
                }]
        }] });
