// @spartacus/organization/esm2015/administration/components/user/form/user-form.service.js
import { Injectable } from '@angular/core';
import { FormArray, FormControl, FormGroup, Validators } from '@angular/forms';
import { B2BUserRole } from '@spartacus/core';
import { CustomFormValidators } from '@spartacus/storefront';
import { FormService } from '../../shared/form/form.service';
import * as i0 from "@angular/core";
export class UserFormService extends FormService {
    build() {
        const form = new FormGroup({});
        form.setControl('customerId', new FormControl(''));
        form.setControl('titleCode', new FormControl(''));
        form.setControl('firstName', new FormControl('', Validators.required));
        form.setControl('lastName', new FormControl('', Validators.required));
        form.setControl('email', new FormControl('', [
            Validators.required,
            CustomFormValidators.emailValidator,
        ]));
        form.setControl('orgUnit', new FormGroup({
            uid: new FormControl(undefined, Validators.required),
        }));
        form.setControl('roles', new FormArray([]));
        form.setControl('isAssignedToApprovers', new FormControl(false));
        form.get('roles').valueChanges.subscribe((roles) => {
            if (roles.includes(B2BUserRole.APPROVER)) {
                form.get('isAssignedToApprovers').enable();
            }
            else {
                form.get('isAssignedToApprovers').disable();
                form.get('isAssignedToApprovers').reset();
            }
        });
        this.form = form;
    }
    patchData(item) {
        var _a;
        super.patchData(item);
        if (item) {
            const roles = this.form.get('roles');
            (_a = item.roles) === null || _a === void 0 ? void 0 : _a.forEach((role) => {
                if (!roles.value.includes(role)) {
                    roles.push(new FormControl(role));
                }
            });
        }
    }
}
UserFormService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserFormService, deps: null, target: i0.ɵɵFactoryTarget.Injectable });
UserFormService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserFormService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserFormService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }] });
