// @spartacus/organization/esm2015/administration/occ/converters/occ-permission-type-list.normalizer.js
import { Injectable } from '@angular/core';
import { PERMISSION_TYPE_NORMALIZER } from '@spartacus/organization/administration/core';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
export class OccPermissionTypeListNormalizer {
    constructor(converter) {
        this.converter = converter;
    }
    convert(source, target) {
        target = source.orderApprovalPermissionTypes.map((permissionType) => this.converter.convert(permissionType, PERMISSION_TYPE_NORMALIZER));
        return target;
    }
}
OccPermissionTypeListNormalizer.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OccPermissionTypeListNormalizer, deps: [{ token: i1.ConverterService }], target: i0.ɵɵFactoryTarget.Injectable });
OccPermissionTypeListNormalizer.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OccPermissionTypeListNormalizer, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OccPermissionTypeListNormalizer, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.ConverterService }]; } });
