// @spartacus/organization/esm2015/administration/core/http-interceptors/bad-request/bad-request.handler.js
import { Injectable } from '@angular/core';
import { GlobalMessageType, HttpErrorHandler, HttpResponseStatus, } from '@spartacus/core';
import * as i0 from "@angular/core";
export class OrganizationBadRequestHandler extends HttpErrorHandler {
    constructor() {
        super(...arguments);
        this.responseStatus = HttpResponseStatus.BAD_REQUEST;
        this.costCenterMask = /ambiguous unique keys \{code\=(.*)\} for model B2BCostCenterModel/;
        this.unitMask = /ambiguous unique keys \{uid\=(.*)\} for model B2BUnitModel/;
        this.permissionMask = /Approval Permission with code\: (.*) already exists\./;
        this.unknownMask = /Model saving error\./;
    }
    hasMatch(errorResponse) {
        return super.hasMatch(errorResponse) && this.matchMask(errorResponse);
    }
    handleError(_request, response) {
        this.getErrors(response).forEach(({ message }) => {
            // Handle cost center conflict
            this.handleOrganizationConflict(message, this.costCenterMask, 'costCenter');
            // Handle unit conflict
            this.handleOrganizationConflict(message, this.unitMask, 'unit');
            // Handle unit conflict
            this.handleOrganizationConflict(message, this.permissionMask, 'permission');
            // Handle unknown conflict
            this.handleOrganizationConflict(message, this.unknownMask, 'unknown');
        });
    }
    matchMask(response) {
        return this.getErrors(response).some((error) => [
            this.costCenterMask,
            this.unitMask,
            this.permissionMask,
            this.unknownMask,
        ].some((mask) => mask.test(error.message)));
    }
    handleOrganizationConflict(message, mask, key) {
        const result = message.match(mask);
        const params = { code: result === null || result === void 0 ? void 0 : result[1] };
        if (result) {
            this.globalMessageService.add({ key: `organization.httpHandlers.conflict.${key}`, params }, GlobalMessageType.MSG_TYPE_ERROR);
        }
    }
    getErrors(response) {
        var _a;
        return (((_a = response.error) === null || _a === void 0 ? void 0 : _a.errors) || []).filter((error) => error.type === 'ModelSavingError' || error.type === 'DuplicateUidError');
    }
    getPriority() {
        return 0 /* NORMAL */;
    }
}
OrganizationBadRequestHandler.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrganizationBadRequestHandler, deps: null, target: i0.ɵɵFactoryTarget.Injectable });
OrganizationBadRequestHandler.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrganizationBadRequestHandler, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrganizationBadRequestHandler, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }] });
