// @spartacus/organization/esm2015/administration/components/shared/card/card.component.js
import { ChangeDetectionStrategy, Component, Input, ViewChild, } from '@angular/core';
import { ViewComponent } from '@spartacus/storefront';
import { tap } from 'rxjs/operators';
import { MessageService } from '../message/services/message.service';
import { ICON_TYPE } from '@spartacus/storefront';
import * as i0 from "@angular/core";
import * as i1 from "../item.service";
import * as i2 from "../message/services/message.service";
import * as i3 from "@spartacus/storefront";
import * as i4 from "../message/message.component";
import * as i5 from "@angular/common";
import * as i6 from "@angular/router";
import * as i7 from "@spartacus/core";
export class CardComponent {
    constructor(itemService, messageService) {
        this.itemService = itemService;
        this.messageService = messageService;
        this.previous = true;
        this.showHint = false;
        this.iconTypes = ICON_TYPE;
        this.item$ = this.itemService.current$.pipe(tap((item) => this.refreshMessages(item)));
    }
    /**
     * The views are router based, which means if we close a view, the router outlet is
     * cleaned immediately. To prevent this, we're closing the view manually, before
     * navigating back.
     */
    closeView(event) {
        event.stopPropagation();
        this.view.toggle(true);
        setTimeout(() => {
            var _a;
            (_a = event.target) === null || _a === void 0 ? void 0 : _a.parentElement.click();
        }, 500);
        return false;
    }
    get previousLabel() {
        return this.previous;
    }
    refreshMessages(item) {
        var _a, _b;
        if (this.itemKey !== undefined &&
            (item === null || item === void 0 ? void 0 : item.code) !== this.itemKey &&
            (item === null || item === void 0 ? void 0 : item.uid) !== this.itemKey &&
            (item === null || item === void 0 ? void 0 : item.customerId) !== this.itemKey) {
            this.messageService.clear();
        }
        this.itemKey = (_b = (_a = item === null || item === void 0 ? void 0 : item.code) !== null && _a !== void 0 ? _a : item === null || item === void 0 ? void 0 : item.uid) !== null && _b !== void 0 ? _b : item === null || item === void 0 ? void 0 : item.customerId;
    }
}
CardComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CardComponent, deps: [{ token: i1.ItemService }, { token: i2.MessageService }], target: i0.ɵɵFactoryTarget.Component });
CardComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: CardComponent, selector: "cx-org-card", inputs: { i18nRoot: "i18nRoot", previous: "previous", subtitle: "subtitle", showHint: "showHint" }, host: { classAttribute: "content-wrapper" }, providers: [MessageService], viewQueries: [{ propertyName: "view", first: true, predicate: ViewComponent, descendants: true, read: ViewComponent }], ngImport: i0, template: "<cx-view class=\"card\">\n  <div class=\"header\">\n    <div class=\"title-bar\">\n      <div class=\"title\">\n        <h3>\n          {{ i18nRoot + '.title' | cxTranslate: { item: item$ | async } }}\n          <button\n            *ngIf=\"showHint\"\n            [cxPopover]=\"detailHint\"\n            [cxPopoverOptions]=\"{\n              placement: 'auto',\n              class: 'hint-popover',\n              appendToBody: true,\n              displayCloseButton: true\n            }\"\n          >\n            <cx-icon [type]=\"iconTypes.INFO\"> </cx-icon>\n          </button>\n        </h3>\n        <h4>\n          {{\n            subtitle ||\n              (i18nRoot + '.subtitle' | cxTranslate: { item: item$ | async })\n          }}\n        </h4>\n      </div>\n      <div class=\"actions\">\n        <ng-content select=\"[actions]\"></ng-content>\n      </div>\n    </div>\n    <a *ngIf=\"!!previous\" class=\"link close\" routerLink=\"../\">\n      <cx-icon\n        *ngIf=\"previous === true; else prevLabel\"\n        type=\"CLOSE\"\n        (click)=\"closeView($event)\"\n      ></cx-icon>\n      <ng-template #prevLabel>{{ previousLabel | cxTranslate }}</ng-template>\n    </a>\n  </div>\n\n  <div class=\"main\">\n    <cx-org-message></cx-org-message>\n    <ng-content select=\"[info]\"></ng-content>\n    <ng-content select=\"[main]\"></ng-content>\n  </div>\n</cx-view>\n\n<router-outlet></router-outlet>\n\n<ng-template #detailHint>\n  <p>\n    {{ i18nRoot + '.hint' | cxTranslate }}\n  </p>\n</ng-template>\n", components: [{ type: i3.ViewComponent, selector: "cx-view", inputs: ["position", "hidden"], outputs: ["hiddenChange"] }, { type: i3.IconComponent, selector: "cx-icon,[cxIcon]", inputs: ["cxIcon", "type"] }, { type: i4.MessageComponent, selector: "cx-org-message" }], directives: [{ type: i5.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i3.PopoverDirective, selector: "[cxPopover]", inputs: ["cxPopover", "cxPopoverOptions"], outputs: ["openPopover", "closePopover"] }, { type: i6.RouterLinkWithHref, selector: "a[routerLink],area[routerLink]", inputs: ["routerLink", "target", "queryParams", "fragment", "queryParamsHandling", "preserveFragment", "skipLocationChange", "replaceUrl", "state", "relativeTo"] }, { type: i6.RouterOutlet, selector: "router-outlet", outputs: ["activate", "deactivate"], exportAs: ["outlet"] }], pipes: { "cxTranslate": i7.TranslatePipe, "async": i5.AsyncPipe }, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CardComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'cx-org-card',
                    templateUrl: './card.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    host: { class: 'content-wrapper' },
                    providers: [MessageService],
                }]
        }], ctorParameters: function () { return [{ type: i1.ItemService }, { type: i2.MessageService }]; }, propDecorators: { i18nRoot: [{
                type: Input
            }], previous: [{
                type: Input
            }], subtitle: [{
                type: Input
            }], showHint: [{
                type: Input
            }], view: [{
                type: ViewChild,
                args: [ViewComponent, { read: ViewComponent }]
            }] } });
