// @spartacus/organization/esm2015/administration/components/shared/card/card.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { I18nModule, UrlModule } from '@spartacus/core';
import { IconModule, SplitViewModule } from '@spartacus/storefront';
import { MessageModule } from '../message/message.module';
import { CardComponent } from './card.component';
import { PopoverModule } from '@spartacus/storefront';
import * as i0 from "@angular/core";
/**
 * Provides a reusable card UI component for the organization split views.
 *
 * The component does not intend to provide a complete set of card features, it's just
 * a reusable component inside the organization UI.
 */
export class CardModule {
}
CardModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CardModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CardModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CardModule, declarations: [CardComponent], imports: [CommonModule,
        SplitViewModule,
        RouterModule,
        I18nModule,
        IconModule,
        UrlModule,
        MessageModule,
        PopoverModule], exports: [CardComponent] });
CardModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CardModule, imports: [[
            CommonModule,
            SplitViewModule,
            RouterModule,
            I18nModule,
            IconModule,
            UrlModule,
            MessageModule,
            PopoverModule,
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CardModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        SplitViewModule,
                        RouterModule,
                        I18nModule,
                        IconModule,
                        UrlModule,
                        MessageModule,
                        PopoverModule,
                    ],
                    declarations: [CardComponent],
                    exports: [CardComponent],
                }]
        }] });
