// @spartacus/organization/esm2015/administration/components/budget/services/budget-item.service.js
import { Injectable } from '@angular/core';
import { ItemService } from '../../shared/item.service';
import * as i0 from "@angular/core";
import * as i1 from "./current-budget.service";
import * as i2 from "@spartacus/core";
import * as i3 from "../form/budget-form.service";
import * as i4 from "@spartacus/organization/administration/core";
export class BudgetItemService extends ItemService {
    constructor(currentItemService, routingService, formService, budgetService) {
        super(currentItemService, routingService, formService);
        this.currentItemService = currentItemService;
        this.routingService = routingService;
        this.formService = formService;
        this.budgetService = budgetService;
    }
    /**
     * @override
     * Returns the budget for the given code.
     *
     * Loads the budget each time, to ensure accurate data is resolved.
     */
    load(code) {
        this.budgetService.loadBudget(code);
        return this.budgetService.get(code);
    }
    update(code, value) {
        this.budgetService.update(code, value);
        return this.budgetService.getLoadingStatus(value.code);
    }
    create(value) {
        this.budgetService.create(value);
        return this.budgetService.getLoadingStatus(value.code);
    }
    /**
     * @override
     * Returns 'budgetDetails'
     */
    getDetailsRoute() {
        return 'orgBudgetDetails';
    }
}
BudgetItemService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BudgetItemService, deps: [{ token: i1.CurrentBudgetService }, { token: i2.RoutingService }, { token: i3.BudgetFormService }, { token: i4.BudgetService }], target: i0.ɵɵFactoryTarget.Injectable });
BudgetItemService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BudgetItemService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BudgetItemService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.CurrentBudgetService }, { type: i2.RoutingService }, { type: i3.BudgetFormService }, { type: i4.BudgetService }]; } });
