// @spartacus/organization/esm2015/administration/components/budget/services/budget-list.service.js
import { Injectable } from '@angular/core';
import { filter, map } from 'rxjs/operators';
import { ListService } from '../../shared/list/list.service';
import { OrganizationTableType } from '../../shared/organization.model';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/storefront";
import * as i2 from "@spartacus/organization/administration/core";
/**
 * Service to populate Budget data to `Table` data. Budget
 * data is driven by the table configuration, using the `OrganizationTables.BUDGET`.
 */
export class BudgetListService extends ListService {
    constructor(tableService, budgetService) {
        super(tableService);
        this.tableService = tableService;
        this.budgetService = budgetService;
        this.tableType = OrganizationTableType.BUDGET;
    }
    load(pagination) {
        return this.budgetService.getList(pagination).pipe(filter((list) => Boolean(list)), map((raw) => this.convertBudgets(raw)));
    }
    /**
     * Populates budget data to a convenient table data model, so that we
     * can skip specific conversion in the view logic where possible.
     */
    convertBudgets({ pagination, sorts, values, }) {
        const budgetModels = {
            pagination,
            sorts,
            values: values.map((value) => {
                var _a;
                return (Object.assign(Object.assign({}, value), { currency: (_a = value.currency) === null || _a === void 0 ? void 0 : _a.isocode, unit: value.orgUnit }));
            }),
        };
        return budgetModels;
    }
}
BudgetListService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BudgetListService, deps: [{ token: i1.TableService }, { token: i2.BudgetService }], target: i0.ɵɵFactoryTarget.Injectable });
BudgetListService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BudgetListService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BudgetListService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.TableService }, { type: i2.BudgetService }]; } });
