// @spartacus/organization/esm2015/administration/components/budget/services/current-budget.service.js
import { Injectable } from '@angular/core';
import { ROUTE_PARAMS } from '@spartacus/organization/administration/root';
import { CurrentItemService } from '../../shared/current-item.service';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
import * as i2 from "@spartacus/organization/administration/core";
export class CurrentBudgetService extends CurrentItemService {
    constructor(routingService, budgetService) {
        super(routingService);
        this.routingService = routingService;
        this.budgetService = budgetService;
    }
    getDetailsRoute() {
        return 'orgBudgetDetails';
    }
    getParamKey() {
        return ROUTE_PARAMS.budgetCode;
    }
    getItem(code) {
        return this.budgetService.get(code);
    }
    getError(code) {
        return this.budgetService.getErrorState(code);
    }
}
CurrentBudgetService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CurrentBudgetService, deps: [{ token: i1.RoutingService }, { token: i2.BudgetService }], target: i0.ɵɵFactoryTarget.Injectable });
CurrentBudgetService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CurrentBudgetService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CurrentBudgetService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.RoutingService }, { type: i2.BudgetService }]; } });
