// @spartacus/organization/esm2015/administration/components/unit/links/addresses/form/unit-address-form.service.js
import { Injectable } from '@angular/core';
import { FormControl, FormGroup, Validators } from '@angular/forms';
import { filter, switchMap, tap } from 'rxjs/operators';
import { FormService } from '../../../../shared/form/form.service';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
import * as i2 from "@spartacus/user/profile/root";
export class UnitAddressFormService extends FormService {
    constructor(userAddressService, userProfileFacade) {
        super();
        this.userAddressService = userAddressService;
        this.userProfileFacade = userProfileFacade;
    }
    build() {
        const form = new FormGroup({});
        form.setControl('id', new FormControl(''));
        form.setControl('titleCode', new FormControl(''));
        form.setControl('firstName', new FormControl('', Validators.required));
        form.setControl('lastName', new FormControl('', Validators.required));
        form.setControl('line1', new FormControl('', Validators.required));
        form.setControl('line2', new FormControl(''));
        form.setControl('town', new FormControl('', Validators.required));
        form.setControl('country', new FormGroup({
            isocode: new FormControl(null, Validators.required),
        }));
        form.setControl('region', new FormGroup({
            isocode: new FormControl(null, Validators.required),
        }));
        form.setControl('postalCode', new FormControl('', Validators.required));
        form.setControl('phone', new FormControl(''));
        this.form = form;
    }
    getCountries() {
        return this.userAddressService.getDeliveryCountries().pipe(tap((countries) => {
            if (Object.keys(countries).length === 0) {
                this.userAddressService.loadDeliveryCountries();
            }
        }));
    }
    getTitles() {
        return this.userProfileFacade.getTitles();
    }
    getRegions() {
        let selectedCountryCode = this.form.get('country.isocode').value;
        let newCountryCode;
        return this.getForm()
            .get('country.isocode')
            .valueChanges.pipe(filter((countryIsoCode) => Boolean(countryIsoCode)), switchMap((countryIsoCode) => {
            newCountryCode = countryIsoCode;
            return this.userAddressService.getRegions(countryIsoCode);
        }), tap((regions) => {
            const regionControl = this.form.get('region.isocode');
            if (!regions || regions.length === 0) {
                regionControl.disable();
            }
            else {
                regionControl.enable();
            }
            if (selectedCountryCode && newCountryCode !== selectedCountryCode) {
                regionControl.reset();
            }
            selectedCountryCode = newCountryCode;
        }));
    }
}
UnitAddressFormService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitAddressFormService, deps: [{ token: i1.UserAddressService }, { token: i2.UserProfileFacade }], target: i0.ɵɵFactoryTarget.Injectable });
UnitAddressFormService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitAddressFormService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitAddressFormService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.UserAddressService }, { type: i2.UserProfileFacade }]; } });
