// @spartacus/organization/esm2015/administration/core/store/selectors/cost-center.selector.js
import { createSelector } from '@ngrx/store';
import { StateUtils, } from '@spartacus/core';
import { COST_CENTER_FEATURE, } from '../organization-state';
import { getBudgetsState } from './budget.selector';
import { getOrganizationState } from './feature.selector';
export const getCostCenterManagementState = createSelector(getOrganizationState, (state) => state[COST_CENTER_FEATURE]);
export const getCostCentersState = createSelector(getCostCenterManagementState, (state) => state && state.entities);
export const getCostCenter = (costCenterCode) => createSelector(getCostCentersState, (state) => StateUtils.entityLoaderStateSelector(state, costCenterCode));
export const getCostCenterValue = (costCenterCode) => {
    return createSelector(getCostCenter(costCenterCode), (costCenterState) => StateUtils.loaderValueSelector(costCenterState));
};
export const getCostCenterList = (params) => createSelector(getCostCenterManagementState, (state) => StateUtils.denormalizeSearch(state, params));
export const getAssignedBudgets = (code, params) => createSelector(getCostCenterManagementState, getBudgetsState, (state, budgets) => StateUtils.denormalizeCustomB2BSearch(state.budgets, budgets, params, code));
export const getCostCenterState = (code) => createSelector(getCostCentersState, (state) => StateUtils.entityLoaderStateSelector(state, code));
