// @spartacus/organization/esm2015/administration/components/shared/message/confirmation/confirmation-message.component.js
import { ChangeDetectionStrategy, Component, Inject, PLATFORM_ID, } from '@angular/core';
import { BaseMessageComponent } from '../base-message.component';
import * as i0 from "@angular/core";
import * as i1 from "../message.model";
import * as i2 from "../services/message.service";
import * as i3 from "@spartacus/storefront";
import * as i4 from "@angular/common";
import * as i5 from "@spartacus/core";
/**
 * Renders a confirmation message and cancel/confirm button in the message component.
 */
export class ConfirmationMessageComponent extends BaseMessageComponent {
    constructor(data, platformId, messageService) {
        super(data, platformId);
        this.data = data;
        this.platformId = platformId;
        this.messageService = messageService;
        this.cancelText = {
            key: 'organization.confirmation.cancel',
        };
        this.confirmText = {
            key: 'organization.confirmation.confirm',
        };
    }
    ngOnInit() {
        var _a, _b;
        super.ngOnInit();
        this.cancelText = (_a = this.messageData.cancel) !== null && _a !== void 0 ? _a : this.cancelText;
        this.confirmText = (_b = this.messageData.confirm) !== null && _b !== void 0 ? _b : this.confirmText;
    }
    /**
     * Emits a confirmation event to the data events.
     *
     * The original author of the event message or other parties can observe
     * the event data.
     */
    confirm() {
        this.data.events.next({ confirm: true });
    }
}
ConfirmationMessageComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ConfirmationMessageComponent, deps: [{ token: i1.MessageData }, { token: PLATFORM_ID }, { token: i2.MessageService }], target: i0.ɵɵFactoryTarget.Component });
ConfirmationMessageComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: ConfirmationMessageComponent, selector: "cx-org-confirmation", usesInheritance: true, ngImport: i0, template: "<div class=\"inner\" [cxFocus]=\"{ focusOnEscape: true }\" (esc)=\"close()\">\n  <p class=\"messageTitle\" *ngIf=\"messageTitle\">\n    {{ messageTitle | cxTranslate }}\n  </p>\n  <div class=\"message\">\n    <cx-icon *ngIf=\"messageIcon\" [type]=\"messageIcon\"></cx-icon>\n    <p>\n      {{ message | cxTranslate }}\n    </p>\n  </div>\n  <div class=\"actions\" [cxFocus]=\"{ autofocus: 'button.primary' }\">\n    <button class=\"button cancel\" (click)=\"close()\">\n      {{ cancelText | cxTranslate }}\n    </button>\n\n    <button class=\"button primary confirm\" (click)=\"confirm()\">\n      {{ confirmText | cxTranslate }}\n    </button>\n  </div>\n</div>\n", components: [{ type: i3.IconComponent, selector: "cx-icon,[cxIcon]", inputs: ["cxIcon", "type"] }], directives: [{ type: i3.FocusDirective, selector: "[cxFocus]", inputs: ["cxFocus"] }, { type: i4.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }], pipes: { "cxTranslate": i5.TranslatePipe }, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: ConfirmationMessageComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'cx-org-confirmation',
                    templateUrl: './confirmation-message.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], ctorParameters: function () { return [{ type: i1.MessageData }, { type: undefined, decorators: [{
                    type: Inject,
                    args: [PLATFORM_ID]
                }] }, { type: i2.MessageService }]; } });
