// @spartacus/organization/esm2015/administration/components/unit/form/unit-form.service.js
import { Injectable } from '@angular/core';
import { FormControl, FormGroup, Validators } from '@angular/forms';
import { CustomFormValidators } from '@spartacus/storefront';
import { FormService } from '../../shared/form/form.service';
import * as i0 from "@angular/core";
export class UnitFormService extends FormService {
    patchData(item) {
        this.toggleParentUnit(item);
        super.patchData(item);
    }
    build() {
        const form = new FormGroup({});
        form.setControl('uid', new FormControl('', [
            Validators.required,
            CustomFormValidators.noSpecialCharacters,
        ]));
        form.setControl('name', new FormControl('', Validators.required));
        form.setControl('approvalProcess', new FormGroup({
            code: new FormControl(null),
        }));
        this.form = form;
        this.toggleParentUnit();
    }
    toggleParentUnit(item) {
        var _a;
        if (this.isRootUnit(item)) {
            (_a = this.form) === null || _a === void 0 ? void 0 : _a.removeControl('parentOrgUnit');
        }
        else if (!this.form.get('parentOrgUnit')) {
            this.form.setControl('parentOrgUnit', new FormGroup({
                uid: new FormControl(null, Validators.required),
            }));
        }
    }
    isRootUnit(item) {
        // as we don't have full response after toggle item status,
        // we have situation where we have object like {uid, active},
        // so decided to check name as alternative required property
        return ((item === null || item === void 0 ? void 0 : item.uid) &&
            (item === null || item === void 0 ? void 0 : item.name) &&
            (!(item === null || item === void 0 ? void 0 : item.parentOrgUnit) || (item === null || item === void 0 ? void 0 : item.uid) === (item === null || item === void 0 ? void 0 : item.parentOrgUnit)));
    }
}
UnitFormService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitFormService, deps: null, target: i0.ɵɵFactoryTarget.Injectable });
UnitFormService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitFormService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitFormService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }] });
