// @spartacus/organization/esm2015/administration/components/user/change-password-form/user-change-password-form.service.js
import { Injectable } from '@angular/core';
import { FormControl, FormGroup, Validators } from '@angular/forms';
import { CustomFormValidators } from '@spartacus/storefront';
import { FormService } from '../../shared/form/form.service';
import * as i0 from "@angular/core";
export class UserChangePasswordFormService extends FormService {
    /**
     * @override
     * Adds the password and confirmPassword field. Also adds the customerId field,
     * so that the customerId can be used during persistent.
     */
    build() {
        const form = new FormGroup({});
        form.setControl('customerId', new FormControl(''));
        form.setControl('password', new FormControl('', [
            Validators.required,
            CustomFormValidators.passwordValidator,
        ]));
        form.setControl('confirmPassword', new FormControl('', [
            Validators.required,
            CustomFormValidators.passwordValidator,
        ]));
        form.setValidators(CustomFormValidators.passwordsMustMatch('password', 'confirmPassword'));
        this.form = form;
    }
    getForm(item) {
        // we need do cleanup, to avoid have filled form after next open of that
        this.form = null;
        return super.getForm(item);
    }
}
UserChangePasswordFormService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserChangePasswordFormService, deps: null, target: i0.ɵɵFactoryTarget.Injectable });
UserChangePasswordFormService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserChangePasswordFormService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserChangePasswordFormService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }] });
