// @spartacus/organization/esm2015/administration/core/connectors/budget/budget.connector.js
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./budget.adapter";
export class BudgetConnector {
    constructor(adapter) {
        this.adapter = adapter;
    }
    get(userId, budgetCode) {
        return this.adapter.load(userId, budgetCode);
    }
    getList(userId, params) {
        return this.adapter.loadList(userId, params);
    }
    create(userId, budget) {
        return this.adapter.create(userId, budget);
    }
    update(userId, budgetCode, budget) {
        return this.adapter.update(userId, budgetCode, budget);
    }
}
BudgetConnector.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BudgetConnector, deps: [{ token: i1.BudgetAdapter }], target: i0.ɵɵFactoryTarget.Injectable });
BudgetConnector.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BudgetConnector, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BudgetConnector, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.BudgetAdapter }]; } });
