// @spartacus/organization/esm2015/administration/components/user/permissions/assigned/user-assigned-permission-list.service.js
import { Injectable } from '@angular/core';
import { map } from 'rxjs/operators';
import { OrganizationTableType } from '../../../shared/organization.model';
import { UserPermissionListService } from '../user-permission-list.service';
import * as i0 from "@angular/core";
export class UserAssignedPermissionListService extends UserPermissionListService {
    constructor() {
        super(...arguments);
        this.tableType = OrganizationTableType.USER_ASSIGNED_PERMISSIONS;
    }
    load(pagination, code) {
        return super
            .load(pagination, code)
            .pipe(map((result) => this.filterSelected(result)));
    }
}
UserAssignedPermissionListService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserAssignedPermissionListService, deps: null, target: i0.ɵɵFactoryTarget.Injectable });
UserAssignedPermissionListService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserAssignedPermissionListService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserAssignedPermissionListService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }] });
