// @spartacus/organization/esm2015/administration/components/user-group/services/current-user-group.service.js
import { Injectable } from '@angular/core';
import { ROUTE_PARAMS } from '@spartacus/organization/administration/root';
import { CurrentItemService } from '../../shared/current-item.service';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
import * as i2 from "@spartacus/organization/administration/core";
export class CurrentUserGroupService extends CurrentItemService {
    constructor(routingService, userGroupService) {
        super(routingService);
        this.routingService = routingService;
        this.userGroupService = userGroupService;
    }
    getParamKey() {
        return ROUTE_PARAMS.userGroupCode;
    }
    getItem(code) {
        return this.userGroupService.get(code);
    }
    getError(code) {
        return this.userGroupService.getErrorState(code);
    }
}
CurrentUserGroupService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CurrentUserGroupService, deps: [{ token: i1.RoutingService }, { token: i2.UserGroupService }], target: i0.ɵɵFactoryTarget.Injectable });
CurrentUserGroupService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CurrentUserGroupService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CurrentUserGroupService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.RoutingService }, { type: i2.UserGroupService }]; } });
