// @spartacus/organization/esm2015/administration/components/user-group/services/user-group-list.service.js
import { Injectable } from '@angular/core';
import { filter, map } from 'rxjs/operators';
import { ListService } from '../../shared/list/list.service';
import { OrganizationTableType } from '../../shared/organization.model';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/storefront";
import * as i2 from "@spartacus/organization/administration/core";
/**
 * Service to populate Cost Center data to `Table` data. The cost center
 * data is driven by the table configuration, using the `OrganizationTables.COST_CENTER`.
 */
export class UserGroupListService extends ListService {
    constructor(tableService, userGroupService) {
        super(tableService);
        this.tableService = tableService;
        this.userGroupService = userGroupService;
        this.tableType = OrganizationTableType.USER_GROUP;
    }
    key() {
        return 'uid';
    }
    load(pagination) {
        return this.userGroupService.getList(pagination).pipe(filter((list) => Boolean(list)), map((raw) => this.convertUserGroups(raw)));
    }
    /**
     * Populates the cost center data to a convenient table data model, so that we
     * can skip specific conversion in the view logic where possible.
     */
    convertUserGroups({ pagination, sorts, values, }) {
        const userGroupModels = {
            pagination,
            sorts,
            values: values.map((value) => (Object.assign(Object.assign({}, value), { unit: value.orgUnit }))),
        };
        return userGroupModels;
    }
}
UserGroupListService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserGroupListService, deps: [{ token: i1.TableService }, { token: i2.UserGroupService }], target: i0.ɵɵFactoryTarget.Injectable });
UserGroupListService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserGroupListService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserGroupListService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.TableService }, { type: i2.UserGroupService }]; } });
