// @spartacus/organization/esm2015/administration/components/user-group/services/user-group-item.service.js
import { Injectable } from '@angular/core';
import { ItemService } from '../../shared/item.service';
import * as i0 from "@angular/core";
import * as i1 from "./current-user-group.service";
import * as i2 from "@spartacus/core";
import * as i3 from "../form/user-group-form.service";
import * as i4 from "@spartacus/organization/administration/core";
export class UserGroupItemService extends ItemService {
    constructor(currentItemService, routingService, formService, userGroupService) {
        super(currentItemService, routingService, formService);
        this.currentItemService = currentItemService;
        this.routingService = routingService;
        this.formService = formService;
        this.userGroupService = userGroupService;
    }
    load(code) {
        this.userGroupService.load(code);
        return this.userGroupService.get(code);
    }
    update(code, value) {
        this.userGroupService.update(code, value);
        return this.userGroupService.getLoadingStatus(value.uid);
    }
    delete(code) {
        this.launchList();
        this.userGroupService.delete(code);
        return this.userGroupService.getLoadingStatus(code);
    }
    create(value) {
        this.userGroupService.create(value);
        return this.userGroupService.getLoadingStatus(value.uid);
    }
    getDetailsRoute() {
        return 'orgUserGroupDetails';
    }
    launchList() {
        this.routingService.go({
            cxRoute: 'orgUserGroup',
        });
    }
}
UserGroupItemService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserGroupItemService, deps: [{ token: i1.CurrentUserGroupService }, { token: i2.RoutingService }, { token: i3.UserGroupFormService }, { token: i4.UserGroupService }], target: i0.ɵɵFactoryTarget.Injectable });
UserGroupItemService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserGroupItemService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserGroupItemService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.CurrentUserGroupService }, { type: i2.RoutingService }, { type: i3.UserGroupFormService }, { type: i4.UserGroupService }]; } });
