// @spartacus/organization/esm2015/administration/components/cost-center/budgets/assigned/cost-center-assigned-budget-list.service.js
import { Injectable } from '@angular/core';
import { map } from 'rxjs/operators';
import { OrganizationTableType } from '../../../shared/organization.model';
import { CostCenterBudgetListService } from '../cost-center-budget-list.service';
import * as i0 from "@angular/core";
export class CostCenterAssignedBudgetListService extends CostCenterBudgetListService {
    constructor() {
        super(...arguments);
        this.tableType = OrganizationTableType.COST_CENTER_ASSIGNED_BUDGETS;
    }
    load(pagination, code) {
        return super
            .load(pagination, code)
            .pipe(map((budgets) => this.filterSelected(budgets)));
    }
}
CostCenterAssignedBudgetListService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CostCenterAssignedBudgetListService, deps: null, target: i0.ɵɵFactoryTarget.Injectable });
CostCenterAssignedBudgetListService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CostCenterAssignedBudgetListService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CostCenterAssignedBudgetListService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }] });
