// @spartacus/organization/esm2015/order-approval/core/store/reducers/index.js
import { InjectionToken } from '@angular/core';
import { combineReducers, } from '@ngrx/store';
import { AuthActions, StateUtils } from '@spartacus/core';
import { ORDER_APPROVAL_ENTITIES, ORDER_APPROVAL_FEATURE, ORDER_APPROVAL_LIST, } from '../order-approval-state';
import { orderApprovalsEntitiesReducer, orderApprovalsListReducer, } from './order-approval.reducer';
export function getReducers() {
    return {
        [ORDER_APPROVAL_FEATURE]: combineReducers({
            entities: StateUtils.entityLoaderReducer(ORDER_APPROVAL_ENTITIES, orderApprovalsEntitiesReducer),
            list: StateUtils.entityLoaderReducer(ORDER_APPROVAL_LIST, orderApprovalsListReducer),
        }),
    };
}
export const reducerToken = new InjectionToken('OrganizationReducers');
export const reducerProvider = {
    provide: reducerToken,
    useFactory: getReducers,
};
export function clearOrganizationState(reducer) {
    return function (state, action) {
        if (action.type === AuthActions.LOGOUT) {
            state = undefined;
        }
        return reducer(state, action);
    };
}
export const metaReducers = [clearOrganizationState];
