// @spartacus/organization/esm2015/administration/components/budget/cost-centers/budget-cost-center-list.component.js
import { ChangeDetectionStrategy, Component } from '@angular/core';
import { ListService } from '../../shared/list/list.service';
import { BudgetCostCenterListService } from './budget-cost-center-list.service';
import * as i0 from "@angular/core";
import * as i1 from "../../shared/sub-list/sub-list.component";
export class BudgetCostCenterListComponent {
}
BudgetCostCenterListComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BudgetCostCenterListComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
BudgetCostCenterListComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: BudgetCostCenterListComponent, selector: "cx-org-budget-cost-center-list", host: { classAttribute: "content-wrapper" }, providers: [
        {
            provide: ListService,
            useExisting: BudgetCostCenterListService,
        },
    ], ngImport: i0, template: "<cx-org-sub-list></cx-org-sub-list>\n", components: [{ type: i1.SubListComponent, selector: "cx-org-sub-list", inputs: ["previous", "key", "showHint", "routerKey"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BudgetCostCenterListComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'cx-org-budget-cost-center-list',
                    templateUrl: './budget-cost-center-list.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    host: { class: 'content-wrapper' },
                    providers: [
                        {
                            provide: ListService,
                            useExisting: BudgetCostCenterListService,
                        },
                    ],
                }]
        }] });
