// @spartacus/organization/esm2015/administration/components/cost-center/details-cell/cost-center-details-cell.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { I18nModule, UrlModule } from '@spartacus/core';
import { PopoverModule } from '@spartacus/storefront';
import { CostCenterDetailsCellComponent } from './cost-center-details-cell.component';
import * as i0 from "@angular/core";
export class CostCenterDetailsCellModule {
}
CostCenterDetailsCellModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CostCenterDetailsCellModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CostCenterDetailsCellModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CostCenterDetailsCellModule, declarations: [CostCenterDetailsCellComponent], imports: [CommonModule, PopoverModule, RouterModule, I18nModule, UrlModule], exports: [CostCenterDetailsCellComponent] });
CostCenterDetailsCellModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CostCenterDetailsCellModule, imports: [[CommonModule, PopoverModule, RouterModule, I18nModule, UrlModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CostCenterDetailsCellModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, PopoverModule, RouterModule, I18nModule, UrlModule],
                    declarations: [CostCenterDetailsCellComponent],
                    exports: [CostCenterDetailsCellComponent],
                }]
        }] });
