// @spartacus/organization/esm2015/administration/components/unit/details-cell/unit-details-cell.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { I18nModule, UrlModule } from '@spartacus/core';
import { PopoverModule } from '@spartacus/storefront';
import { UnitDetailsCellComponent } from './unit-details-cell.component';
import * as i0 from "@angular/core";
export class UnitDetailsCellModule {
}
UnitDetailsCellModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitDetailsCellModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
UnitDetailsCellModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitDetailsCellModule, declarations: [UnitDetailsCellComponent], imports: [CommonModule, PopoverModule, RouterModule, I18nModule, UrlModule], exports: [UnitDetailsCellComponent] });
UnitDetailsCellModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitDetailsCellModule, imports: [[CommonModule, PopoverModule, RouterModule, I18nModule, UrlModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitDetailsCellModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, PopoverModule, RouterModule, I18nModule, UrlModule],
                    declarations: [UnitDetailsCellComponent],
                    exports: [UnitDetailsCellComponent],
                }]
        }] });
