// @spartacus/organization/esm2015/administration/components/unit/links/addresses/list/unit-address-list.service.js
import { Injectable } from '@angular/core';
import { SubListService } from '../../../../shared/sub-list/sub-list.service';
import { OrganizationTableType } from '../../../../shared/organization.model';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/storefront";
import * as i2 from "@spartacus/organization/administration/core";
export class UnitAddressListService extends SubListService {
    constructor(tableService, orgUnitService) {
        super(tableService);
        this.tableService = tableService;
        this.orgUnitService = orgUnitService;
        this.tableType = OrganizationTableType.UNIT_ADDRESS;
        this._domainType = OrganizationTableType.UNIT_ADDRESS;
    }
    load(_pagination, code) {
        return this.orgUnitService.getAddresses(code);
    }
}
UnitAddressListService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitAddressListService, deps: [{ token: i1.TableService }, { token: i2.OrgUnitService }], target: i0.ɵɵFactoryTarget.Injectable });
UnitAddressListService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitAddressListService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitAddressListService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.TableService }, { type: i2.OrgUnitService }]; } });
