// @spartacus/organization/esm2015/order-approval/core/order-approval-core.module.js
import { NgModule } from '@angular/core';
import { OrderApprovalStoreModule } from './store/order-approval-store.module';
import { OrderApprovalConnector } from './connectors/order-approval.connector';
import * as i0 from "@angular/core";
export class OrderApprovalCoreModule {
    static forRoot() {
        return {
            ngModule: OrderApprovalCoreModule,
            providers: [OrderApprovalConnector],
        };
    }
}
OrderApprovalCoreModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderApprovalCoreModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
OrderApprovalCoreModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderApprovalCoreModule, imports: [OrderApprovalStoreModule] });
OrderApprovalCoreModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderApprovalCoreModule, imports: [[OrderApprovalStoreModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: OrderApprovalCoreModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [OrderApprovalStoreModule],
                }]
        }] });
