// @spartacus/organization/esm2015/administration/components/user-group/users/user-group-user-list.component.js
import { ChangeDetectionStrategy, Component, ViewChild } from '@angular/core';
import { filter, first, switchMap, take } from 'rxjs/operators';
import { LoadStatus, } from '@spartacus/organization/administration/core';
import { ListService } from '../../shared/list/list.service';
import { UserGroupUserListService } from './user-group-user-list.service';
import * as i0 from "@angular/core";
import * as i1 from "../services/current-user-group.service";
import * as i2 from "./user-group-user-list.service";
import * as i3 from "../../shared/sub-list/sub-list.component";
import * as i4 from "@angular/router";
import * as i5 from "@spartacus/core";
export class UserGroupUserListComponent {
    constructor(currentUserGroupService, userGroupUserListService) {
        this.currentUserGroupService = currentUserGroupService;
        this.userGroupUserListService = userGroupUserListService;
    }
    unassignAll() {
        this.currentUserGroupService.key$
            .pipe(first(), switchMap((key) => this.userGroupUserListService.unassignAllMembers(key).pipe(take(1), filter((data) => data.status === LoadStatus.SUCCESS))))
            .subscribe((data) => {
            this.notify(data.item);
        });
    }
    notify(item) {
        this.subList.messageService.add({
            message: {
                key: `orgUserGroupUsers.unassignAllConfirmation`,
                params: {
                    item,
                },
            },
        });
    }
}
UserGroupUserListComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserGroupUserListComponent, deps: [{ token: i1.CurrentUserGroupService }, { token: i2.UserGroupUserListService }], target: i0.ɵɵFactoryTarget.Component });
UserGroupUserListComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: UserGroupUserListComponent, selector: "cx-org-user-group-user-list", host: { classAttribute: "content-wrapper" }, providers: [
        {
            provide: ListService,
            useExisting: UserGroupUserListService,
        },
    ], viewQueries: [{ propertyName: "subList", first: true, predicate: ["subList"], descendants: true }], ngImport: i0, template: "<cx-org-sub-list [previous]=\"false\" #subList>\n  <button actions (click)=\"unassignAll()\" class=\"link\">\n    {{ 'orgUserGroupUsers.unassignAll' | cxTranslate }}\n  </button>\n  <a actions class=\"link\" routerLink=\"../\">\n    {{ 'organization.done' | cxTranslate }}\n  </a>\n</cx-org-sub-list>\n", components: [{ type: i3.SubListComponent, selector: "cx-org-sub-list", inputs: ["previous", "key", "showHint", "routerKey"] }], directives: [{ type: i4.RouterLinkWithHref, selector: "a[routerLink],area[routerLink]", inputs: ["routerLink", "target", "queryParams", "fragment", "queryParamsHandling", "preserveFragment", "skipLocationChange", "replaceUrl", "state", "relativeTo"] }], pipes: { "cxTranslate": i5.TranslatePipe }, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserGroupUserListComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'cx-org-user-group-user-list',
                    templateUrl: './user-group-user-list.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    host: { class: 'content-wrapper' },
                    providers: [
                        {
                            provide: ListService,
                            useExisting: UserGroupUserListService,
                        },
                    ],
                }]
        }], ctorParameters: function () { return [{ type: i1.CurrentUserGroupService }, { type: i2.UserGroupUserListService }]; }, propDecorators: { subList: [{
                type: ViewChild,
                args: ['subList']
            }] } });
