// @spartacus/organization/esm2015/administration/components/user-group/users/user-group-user-list.service.js
import { Injectable } from '@angular/core';
import { SubListService } from '../../shared/sub-list/sub-list.service';
import { OrganizationTableType } from '../../shared/organization.model';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/storefront";
import * as i2 from "@spartacus/organization/administration/core";
export class UserGroupUserListService extends SubListService {
    constructor(tableService, userGroupService, userService) {
        super(tableService);
        this.tableService = tableService;
        this.userGroupService = userGroupService;
        this.userService = userService;
        this.tableType = OrganizationTableType.USER_GROUP_USERS;
        this._domainType = OrganizationTableType.USER;
    }
    /**
     *
     * @override
     * Loads all b2b users.
     *
     * @param code The user group code.
     */
    load(pagination, code) {
        return this.userGroupService.getAvailableOrgCustomers(code, pagination);
    }
    /**
     * @override
     * Assign user to the user group.
     */
    assign(userGroupCode, customerId) {
        this.userGroupService.assignMember(userGroupCode, customerId);
        return this.userService.getLoadingStatus(customerId);
    }
    /**
     * @override
     * Unassigns the user from the user group.
     */
    unassign(userGroupCode, customerId) {
        this.userGroupService.unassignMember(userGroupCode, customerId);
        return this.userService.getLoadingStatus(customerId);
    }
    unassignAllMembers(userGroupCode) {
        this.userGroupService.unassignAllMembers(userGroupCode);
        return this.userGroupService.getLoadingStatus(userGroupCode);
    }
}
UserGroupUserListService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserGroupUserListService, deps: [{ token: i1.TableService }, { token: i2.UserGroupService }, { token: i2.B2BUserService }], target: i0.ɵɵFactoryTarget.Injectable });
UserGroupUserListService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserGroupUserListService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UserGroupUserListService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.TableService }, { type: i2.UserGroupService }, { type: i2.B2BUserService }]; } });
