// @spartacus/organization/esm2015/administration/components/cost-center/budgets/cost-center-budget-list.service.js
import { Injectable } from '@angular/core';
import { SubListService } from '../../shared/sub-list/sub-list.service';
import { OrganizationTableType } from '../../shared/organization.model';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/storefront";
import * as i2 from "@spartacus/organization/administration/core";
export class CostCenterBudgetListService extends SubListService {
    constructor(tableService, costCenterService, budgetService) {
        super(tableService);
        this.tableService = tableService;
        this.costCenterService = costCenterService;
        this.budgetService = budgetService;
        this.tableType = OrganizationTableType.COST_CENTER_BUDGETS;
        this._domainType = OrganizationTableType.BUDGET;
    }
    load(pagination, code) {
        return this.costCenterService.getBudgets(code, pagination);
    }
    /**
     * @override
     * Assign budget to the cost center.
     */
    assign(costCenterCode, budgetCode) {
        this.costCenterService.assignBudget(costCenterCode, budgetCode);
        return this.budgetService.getLoadingStatus(budgetCode);
    }
    /**
     * @override
     * Unassign the budget from the cost center.
     */
    unassign(costCenterCode, budgetCode) {
        this.costCenterService.unassignBudget(costCenterCode, budgetCode);
        return this.budgetService.getLoadingStatus(budgetCode);
    }
}
CostCenterBudgetListService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CostCenterBudgetListService, deps: [{ token: i1.TableService }, { token: i2.CostCenterService }, { token: i2.BudgetService }], target: i0.ɵɵFactoryTarget.Injectable });
CostCenterBudgetListService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CostCenterBudgetListService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CostCenterBudgetListService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.TableService }, { type: i2.CostCenterService }, { type: i2.BudgetService }]; } });
