// @spartacus/organization/esm2015/administration/components/shared/detail/disable-info/disable-info.service.js
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
export class DisableInfoService {
    isItemDisabled(item) {
        return (item === null || item === void 0 ? void 0 : item.active) === false;
    }
    isParentDisabled(item) {
        var _a;
        return (((_a = (item.orgUnit || item.unit || item.parentOrgUnit)) === null || _a === void 0 ? void 0 : _a.active) === false &&
            !this.isRootUnit(item));
    }
    isRootUnit(item) {
        var _a, _b;
        return ((item === null || item === void 0 ? void 0 : item.uid) &&
            (item === null || item === void 0 ? void 0 : item.name) &&
            !((_a = item) === null || _a === void 0 ? void 0 : _a.orgUnit) &&
            !((_b = item) === null || _b === void 0 ? void 0 : _b.unit) &&
            (!(item === null || item === void 0 ? void 0 : item.parentOrgUnit) || (item === null || item === void 0 ? void 0 : item.uid) === (item === null || item === void 0 ? void 0 : item.parentOrgUnit)));
    }
}
DisableInfoService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: DisableInfoService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
DisableInfoService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: DisableInfoService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: DisableInfoService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }] });
