// @spartacus/organization/esm2015/administration/components/shared/detail/disable-info/disable-info.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { DisableInfoComponent } from './disable-info.component';
import { IconModule } from '@spartacus/storefront';
import { I18nModule } from '@spartacus/core';
import * as i0 from "@angular/core";
export class DisableInfoModule {
}
DisableInfoModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: DisableInfoModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
DisableInfoModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: DisableInfoModule, declarations: [DisableInfoComponent], imports: [CommonModule, IconModule, I18nModule], exports: [DisableInfoComponent] });
DisableInfoModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: DisableInfoModule, imports: [[CommonModule, IconModule, I18nModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: DisableInfoModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, IconModule, I18nModule],
                    declarations: [DisableInfoComponent],
                    exports: [DisableInfoComponent],
                }]
        }] });
