// @spartacus/organization/esm2015/administration/components/budget/form/budget-form.service.js
import { Injectable } from '@angular/core';
import { FormControl, FormGroup, Validators } from '@angular/forms';
import { CustomFormValidators } from '@spartacus/storefront';
import { FormService } from '../../shared/form/form.service';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/storefront";
export class BudgetFormService extends FormService {
    constructor(datePickerService) {
        super();
        this.datePickerService = datePickerService;
    }
    build() {
        const form = new FormGroup({});
        form.setControl('code', new FormControl('', [
            Validators.required,
            CustomFormValidators.noSpecialCharacters,
        ]));
        form.setControl('name', new FormControl('', Validators.required));
        form.setControl('startDate', new FormControl('', [
            Validators.required,
            CustomFormValidators.patternValidation((date) => this.datePickerService.isValidFormat(date)),
        ]));
        form.setControl('endDate', new FormControl('', [
            Validators.required,
            CustomFormValidators.patternValidation((date) => this.datePickerService.isValidFormat(date)),
        ]));
        form.setControl('budget', new FormControl('', [
            Validators.required,
            CustomFormValidators.mustBePositive,
        ]));
        form.setControl('currency', new FormGroup({
            isocode: new FormControl(undefined, Validators.required),
        }));
        form.setControl('orgUnit', new FormGroup({
            uid: new FormControl(undefined, Validators.required),
        }));
        form.setValidators(CustomFormValidators.dateRange('startDate', 'endDate', (date) => this.datePickerService.getDate(date)));
        this.form = form;
    }
}
BudgetFormService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BudgetFormService, deps: [{ token: i1.DatePickerService }], target: i0.ɵɵFactoryTarget.Injectable });
BudgetFormService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BudgetFormService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: BudgetFormService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.DatePickerService }]; } });
