// @spartacus/organization/esm2015/administration/components/permission/services/current-permission.service.js
import { Injectable } from '@angular/core';
import { ROUTE_PARAMS } from '@spartacus/organization/administration/root';
import { CurrentItemService } from '../../shared/current-item.service';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/core";
import * as i2 from "@spartacus/organization/administration/core";
export class CurrentPermissionService extends CurrentItemService {
    constructor(routingService, permissionService) {
        super(routingService);
        this.routingService = routingService;
        this.permissionService = permissionService;
    }
    getParamKey() {
        return ROUTE_PARAMS.permissionCode;
    }
    getItem(code) {
        return this.permissionService.get(code);
    }
    getError(code) {
        return this.permissionService.getErrorState(code);
    }
}
CurrentPermissionService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CurrentPermissionService, deps: [{ token: i1.RoutingService }, { token: i2.PermissionService }], target: i0.ɵɵFactoryTarget.Injectable });
CurrentPermissionService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CurrentPermissionService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: CurrentPermissionService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.RoutingService }, { type: i2.PermissionService }]; } });
