// @spartacus/organization/esm2015/administration/components/permission/services/permission-list.service.js
import { Injectable } from '@angular/core';
import { filter, map } from 'rxjs/operators';
import { ListService } from '../../shared/list/list.service';
import { OrganizationTableType } from '../../shared/organization.model';
import * as i0 from "@angular/core";
import * as i1 from "@spartacus/storefront";
import * as i2 from "@spartacus/organization/administration/core";
/**
 * Service to populate permission data to `Table` data. The permission
 * data is driven by the table configuration, using the `OrganizationTables.PERMISSION`.
 */
export class PermissionListService extends ListService {
    constructor(tableService, permissionsService) {
        super(tableService);
        this.tableService = tableService;
        this.permissionsService = permissionsService;
        this.tableType = OrganizationTableType.PERMISSION;
    }
    load(pagination) {
        return this.permissionsService.getList(pagination).pipe(filter((list) => Boolean(list)), map((raw) => this.convertPermissions(raw)));
    }
    /**
     * Populates the permission data to a convenient table data model, so that we
     * can skip specific conversion in the view logic where possible.
     */
    convertPermissions({ pagination, sorts, values, }) {
        const permissionGroupModels = {
            pagination,
            sorts,
            values: values.map((value) => (Object.assign(Object.assign({}, value), { unit: value.orgUnit }))),
        };
        return permissionGroupModels;
    }
}
PermissionListService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PermissionListService, deps: [{ token: i1.TableService }, { token: i2.PermissionService }], target: i0.ɵɵFactoryTarget.Injectable });
PermissionListService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PermissionListService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PermissionListService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.TableService }, { type: i2.PermissionService }]; } });
