// @spartacus/organization/esm2015/administration/components/permission/services/permission-item.service.js
import { Injectable } from '@angular/core';
import { ItemService } from '../../shared/item.service';
import * as i0 from "@angular/core";
import * as i1 from "./current-permission.service";
import * as i2 from "@spartacus/core";
import * as i3 from "../form/permission-form.service";
import * as i4 from "@spartacus/organization/administration/core";
export class PermissionItemService extends ItemService {
    constructor(currentItemService, routingService, formService, permissionService) {
        super(currentItemService, routingService, formService);
        this.currentItemService = currentItemService;
        this.routingService = routingService;
        this.formService = formService;
        this.permissionService = permissionService;
    }
    load(code) {
        this.permissionService.loadPermission(code);
        return this.permissionService.get(code);
    }
    update(code, value) {
        this.permissionService.update(code, value);
        return this.permissionService.getLoadingStatus(value.code);
    }
    create(value) {
        this.permissionService.create(value);
        return this.permissionService.getLoadingStatus(value.code);
    }
    getDetailsRoute() {
        return 'orgPurchaseLimitDetails';
    }
}
PermissionItemService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PermissionItemService, deps: [{ token: i1.CurrentPermissionService }, { token: i2.RoutingService }, { token: i3.PermissionFormService }, { token: i4.PermissionService }], target: i0.ɵɵFactoryTarget.Injectable });
PermissionItemService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PermissionItemService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: PermissionItemService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.CurrentPermissionService }, { type: i2.RoutingService }, { type: i3.PermissionFormService }, { type: i4.PermissionService }]; } });
