// @spartacus/organization/bundles/spartacus-organization-order-approval-assets.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports) :
    typeof define === 'function' && define.amd ? define('@spartacus/organization/order-approval/assets', ['exports'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.spartacus = global.spartacus || {}, global.spartacus.organization = global.spartacus.organization || {}, global.spartacus.organization['order-approval'] = global.spartacus.organization['order-approval'] || {}, global.spartacus.organization['order-approval'].assets = {})));
}(this, (function (exports) { 'use strict';

    var orderApprovalDetails = {
        back: 'Back To List',
        showForm_APPROVE: 'Approve Order...',
        showForm_REJECT: 'Reject Order...',
        form: {
            title_APPROVE: 'Approve order {{orderCode}} totalling {{orderTotal}}',
            title_REJECT: 'Reject order {{orderCode}} totalling {{orderTotal}}',
            submit_APPROVE: 'Approve',
            submit_REJECT: 'Reject',
            cancel: 'Cancel',
            comment_APPROVE: {
                label: 'Comment (optional, maximum 255 characters)',
                placeholder: '',
            },
            comment_REJECT: {
                label: 'Comment (maximum 255 characters)',
                placeholder: '',
            },
        },
        permissionResults: {
            header: 'Customer Approval Details',
            permission: 'Permission',
            approver: 'Approver',
            status: 'Status',
            approverComments: 'Approver Comments',
            noApprovalComments: 'None',
            permissionType_B2BBudgetExceededPermission: 'The budget associated with the chosen cost center has been exceeded',
            permissionType_B2BOrderThresholdPermission: 'Order total exceeded the per-order limit',
            permissionType_B2BOrderThresholdTimespanPermission: 'Order total exceeded the per-timespan limit',
        },
    };
    var orderApprovalList = {
        orderCode: 'Order #',
        POCode: 'PO #',
        placedBy: 'Placed by',
        date: 'Date',
        status: 'Status',
        total: 'Total',
        none: 'None',
        emptyList: 'There are no orders to approve at this time.',
    };
    var orderApprovalGlobal = {
        notification: {
            noSufficientPermissions: 'No sufficient permissions to access this page.',
        },
    };
    var orderApproval = {
        orderApprovalDetails: orderApprovalDetails,
        orderApprovalList: orderApprovalList,
        orderApprovalGlobal: orderApprovalGlobal,
    };

    var en = {
        orderApproval: orderApproval,
    };

    var orderApprovalTranslations = {
        en: en,
    };
    // expose all translation chunk mapping for orderApproval feature
    var orderApprovalTranslationChunksConfig = {
        orderApproval: [
            'orderApprovalDetails',
            'orderApprovalList',
            'orderApprovalGlobal',
        ],
    };

    /**
     * Generated bundle index. Do not edit.
     */

    exports.orderApprovalTranslationChunksConfig = orderApprovalTranslationChunksConfig;
    exports.orderApprovalTranslations = orderApprovalTranslations;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
