// @spartacus/organization/bundles/spartacus-organization-order-approval-root.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/router'), require('@spartacus/core'), require('@spartacus/storefront')) :
    typeof define === 'function' && define.amd ? define('@spartacus/organization/order-approval/root', ['exports', '@angular/core', '@angular/router', '@spartacus/core', '@spartacus/storefront'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.spartacus = global.spartacus || {}, global.spartacus.organization = global.spartacus.organization || {}, global.spartacus.organization['order-approval'] = global.spartacus.organization['order-approval'] || {}, global.spartacus.organization['order-approval'].root = {}), global.ng.core, global.ng.router, global.core, global.storefront));
}(this, (function (exports, i0, i1, core, storefront) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () {
                            return e[k];
                        }
                    });
                }
            });
        }
        n['default'] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);

    var ORGANIZATION_ORDER_APPROVAL_FEATURE = 'organizationOrderApproval';

    // TODO: Inline this factory when we start releasing Ivy compiled libraries
    function defaultOrganizationOrderApprovalComponentsConfig() {
        var _a;
        var config = {
            featureModules: (_a = {},
                _a[ORGANIZATION_ORDER_APPROVAL_FEATURE] = {
                    cmsComponents: [
                        'OrderApprovalListComponent',
                        'OrderApprovalDetailTotalsComponent',
                        'OrderApprovalDetailApprovalDetailsComponent',
                        'OrderApprovalDetailShippingComponent',
                        'OrderApprovalDetailItemsComponent',
                        'OrderApprovalDetailFormComponent',
                        'AccountOrderDetailsApprovalDetailsComponent',
                    ],
                },
                _a),
        };
        return config;
    }
    var OrderApprovalRootModule = /** @class */ (function () {
        function OrderApprovalRootModule() {
        }
        return OrderApprovalRootModule;
    }());
    OrderApprovalRootModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: OrderApprovalRootModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    OrderApprovalRootModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: OrderApprovalRootModule, imports: [i1__namespace.RouterModule] });
    OrderApprovalRootModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: OrderApprovalRootModule, providers: [
            core.provideDefaultConfigFactory(defaultOrganizationOrderApprovalComponentsConfig),
            core.provideDefaultConfig({
                routing: {
                    routes: {
                        orderApprovals: {
                            paths: ['my-account/approval-dashboard'],
                        },
                        orderApprovalDetails: {
                            paths: ['my-account/approval/:approvalCode'],
                            paramsMapping: { approvalCode: 'approvalCode' },
                        },
                    },
                },
            }),
        ], imports: [[
                i1.RouterModule.forChild([
                    {
                        path: null,
                        canActivate: [core.AuthGuard, storefront.CmsPageGuard],
                        component: storefront.PageLayoutComponent,
                        data: { cxRoute: 'orderApprovalDetails' },
                    },
                ]),
            ]] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0__namespace, type: OrderApprovalRootModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        imports: [
                            i1.RouterModule.forChild([
                                {
                                    path: null,
                                    canActivate: [core.AuthGuard, storefront.CmsPageGuard],
                                    component: storefront.PageLayoutComponent,
                                    data: { cxRoute: 'orderApprovalDetails' },
                                },
                            ]),
                        ],
                        providers: [
                            core.provideDefaultConfigFactory(defaultOrganizationOrderApprovalComponentsConfig),
                            core.provideDefaultConfig({
                                routing: {
                                    routes: {
                                        orderApprovals: {
                                            paths: ['my-account/approval-dashboard'],
                                        },
                                        orderApprovalDetails: {
                                            paths: ['my-account/approval/:approvalCode'],
                                            paramsMapping: { approvalCode: 'approvalCode' },
                                        },
                                    },
                                },
                            }),
                        ],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.ORGANIZATION_ORDER_APPROVAL_FEATURE = ORGANIZATION_ORDER_APPROVAL_FEATURE;
    exports.OrderApprovalRootModule = OrderApprovalRootModule;
    exports.defaultOrganizationOrderApprovalComponentsConfig = defaultOrganizationOrderApprovalComponentsConfig;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
