// @spartacus/organization/esm2015/administration/components/unit/links/children/create/unit-child-item.service.js
import { Injectable } from '@angular/core';
import { UnitItemService } from '../../../services/unit-item.service';
import * as i0 from "@angular/core";
import * as i1 from "./current-unit-child.service";
import * as i2 from "@spartacus/core";
import * as i3 from "../../../form/unit-form.service";
import * as i4 from "@spartacus/organization/administration/core";
export class UnitChildItemService extends UnitItemService {
    constructor(currentItemService, routingService, formService, unitService) {
        super(currentItemService, routingService, formService, unitService);
        this.currentItemService = currentItemService;
        this.routingService = routingService;
        this.formService = formService;
        this.unitService = unitService;
    }
    save(form, key) {
        var _a;
        // we enable the parentOrgUnit temporarily so that the underlying
        // save method can read the complete form.value.
        (_a = form.get('parentOrgUnit')) === null || _a === void 0 ? void 0 : _a.enable();
        return super.save(form, key);
    }
    /**
     * @override
     * Returns 'unitDetails'
     */
    getDetailsRoute() {
        return 'orgUnitChildren';
    }
    buildRouteParams(item) {
        return { uid: item.parentOrgUnit.uid };
    }
}
UnitChildItemService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitChildItemService, deps: [{ token: i1.CurrentUnitChildService }, { token: i2.RoutingService }, { token: i3.UnitFormService }, { token: i4.OrgUnitService }], target: i0.ɵɵFactoryTarget.Injectable });
UnitChildItemService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitChildItemService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitChildItemService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.CurrentUnitChildService }, { type: i2.RoutingService }, { type: i3.UnitFormService }, { type: i4.OrgUnitService }]; } });
