// @spartacus/organization/esm2015/administration/components/unit/links/children/create/unit-child-create.component.js
import { ChangeDetectionStrategy, Component } from '@angular/core';
import { UnitItemService } from '../../../services/unit-item.service';
import { UnitChildItemService } from './unit-child-item.service';
import * as i0 from "@angular/core";
import * as i1 from "../../../services/current-unit.service";
import * as i2 from "../../../form/unit-form.component";
export class UnitChildCreateComponent {
    constructor(unitService) {
        this.unitService = unitService;
        this.unitKey$ = this.unitService.key$;
    }
}
UnitChildCreateComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitChildCreateComponent, deps: [{ token: i1.CurrentUnitService }], target: i0.ɵɵFactoryTarget.Component });
UnitChildCreateComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.0.5", type: UnitChildCreateComponent, selector: "cx-org-unit-child-create", host: { classAttribute: "content-wrapper" }, providers: [
        // we provide a specific version of the `UnitItemService` to
        // let the form component work with child units.
        {
            provide: UnitItemService,
            useExisting: UnitChildItemService,
        },
    ], ngImport: i0, template: "<cx-org-unit-form\n  [createChildUnit]=\"true\"\n  i18nRoot=\"orgUnit.children\"\n></cx-org-unit-form>\n", components: [{ type: i2.UnitFormComponent, selector: "cx-org-unit-form", inputs: ["i18nRoot", "createChildUnit"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.5", ngImport: i0, type: UnitChildCreateComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'cx-org-unit-child-create',
                    templateUrl: './unit-child-create.component.html',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    host: { class: 'content-wrapper' },
                    providers: [
                        // we provide a specific version of the `UnitItemService` to
                        // let the form component work with child units.
                        {
                            provide: UnitItemService,
                            useExisting: UnitChildItemService,
                        },
                    ],
                }]
        }], ctorParameters: function () { return [{ type: i1.CurrentUnitService }]; } });
